// @aws-sdk/client-lambda/dist-es/models/models_0.js
import { __assign, __extends } from "tslib";
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { LambdaServiceException as __BaseException } from "./LambdaServiceException";
export var AccountLimit;
(function (AccountLimit) {
    AccountLimit.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AccountLimit || (AccountLimit = {}));
export var AccountUsage;
(function (AccountUsage) {
    AccountUsage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AccountUsage || (AccountUsage = {}));
export var AddLayerVersionPermissionRequest;
(function (AddLayerVersionPermissionRequest) {
    AddLayerVersionPermissionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddLayerVersionPermissionRequest || (AddLayerVersionPermissionRequest = {}));
export var AddLayerVersionPermissionResponse;
(function (AddLayerVersionPermissionResponse) {
    AddLayerVersionPermissionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddLayerVersionPermissionResponse || (AddLayerVersionPermissionResponse = {}));
var InvalidParameterValueException = (function (_super) {
    __extends(InvalidParameterValueException, _super);
    function InvalidParameterValueException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidParameterValueException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidParameterValueException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidParameterValueException.prototype);
        _this.Type = opts.Type;
        return _this;
    }
    return InvalidParameterValueException;
}(__BaseException));
export { InvalidParameterValueException };
var PolicyLengthExceededException = (function (_super) {
    __extends(PolicyLengthExceededException, _super);
    function PolicyLengthExceededException(opts) {
        var _this = _super.call(this, __assign({ name: "PolicyLengthExceededException", $fault: "client" }, opts)) || this;
        _this.name = "PolicyLengthExceededException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, PolicyLengthExceededException.prototype);
        _this.Type = opts.Type;
        return _this;
    }
    return PolicyLengthExceededException;
}(__BaseException));
export { PolicyLengthExceededException };
var PreconditionFailedException = (function (_super) {
    __extends(PreconditionFailedException, _super);
    function PreconditionFailedException(opts) {
        var _this = _super.call(this, __assign({ name: "PreconditionFailedException", $fault: "client" }, opts)) || this;
        _this.name = "PreconditionFailedException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, PreconditionFailedException.prototype);
        _this.Type = opts.Type;
        return _this;
    }
    return PreconditionFailedException;
}(__BaseException));
export { PreconditionFailedException };
var ResourceConflictException = (function (_super) {
    __extends(ResourceConflictException, _super);
    function ResourceConflictException(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceConflictException", $fault: "client" }, opts)) || this;
        _this.name = "ResourceConflictException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceConflictException.prototype);
        _this.Type = opts.Type;
        return _this;
    }
    return ResourceConflictException;
}(__BaseException));
export { ResourceConflictException };
var ResourceNotFoundException = (function (_super) {
    __extends(ResourceNotFoundException, _super);
    function ResourceNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "ResourceNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceNotFoundException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return ResourceNotFoundException;
}(__BaseException));
export { ResourceNotFoundException };
var ServiceException = (function (_super) {
    __extends(ServiceException, _super);
    function ServiceException(opts) {
        var _this = _super.call(this, __assign({ name: "ServiceException", $fault: "server" }, opts)) || this;
        _this.name = "ServiceException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, ServiceException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return ServiceException;
}(__BaseException));
export { ServiceException };
export var ThrottleReason;
(function (ThrottleReason) {
    ThrottleReason["CallerRateLimitExceeded"] = "CallerRateLimitExceeded";
    ThrottleReason["ConcurrentInvocationLimitExceeded"] = "ConcurrentInvocationLimitExceeded";
    ThrottleReason["FunctionInvocationRateLimitExceeded"] = "FunctionInvocationRateLimitExceeded";
    ThrottleReason["ReservedFunctionConcurrentInvocationLimitExceeded"] = "ReservedFunctionConcurrentInvocationLimitExceeded";
    ThrottleReason["ReservedFunctionInvocationRateLimitExceeded"] = "ReservedFunctionInvocationRateLimitExceeded";
})(ThrottleReason || (ThrottleReason = {}));
var TooManyRequestsException = (function (_super) {
    __extends(TooManyRequestsException, _super);
    function TooManyRequestsException(opts) {
        var _this = _super.call(this, __assign({ name: "TooManyRequestsException", $fault: "client" }, opts)) || this;
        _this.name = "TooManyRequestsException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, TooManyRequestsException.prototype);
        _this.retryAfterSeconds = opts.retryAfterSeconds;
        _this.Type = opts.Type;
        _this.Reason = opts.Reason;
        return _this;
    }
    return TooManyRequestsException;
}(__BaseException));
export { TooManyRequestsException };
export var FunctionUrlAuthType;
(function (FunctionUrlAuthType) {
    FunctionUrlAuthType["AWS_IAM"] = "AWS_IAM";
    FunctionUrlAuthType["NONE"] = "NONE";
})(FunctionUrlAuthType || (FunctionUrlAuthType = {}));
export var AddPermissionRequest;
(function (AddPermissionRequest) {
    AddPermissionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddPermissionRequest || (AddPermissionRequest = {}));
export var AddPermissionResponse;
(function (AddPermissionResponse) {
    AddPermissionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddPermissionResponse || (AddPermissionResponse = {}));
export var AliasRoutingConfiguration;
(function (AliasRoutingConfiguration) {
    AliasRoutingConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AliasRoutingConfiguration || (AliasRoutingConfiguration = {}));
export var AliasConfiguration;
(function (AliasConfiguration) {
    AliasConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AliasConfiguration || (AliasConfiguration = {}));
export var AllowedPublishers;
(function (AllowedPublishers) {
    AllowedPublishers.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AllowedPublishers || (AllowedPublishers = {}));
export var Architecture;
(function (Architecture) {
    Architecture["arm64"] = "arm64";
    Architecture["x86_64"] = "x86_64";
})(Architecture || (Architecture = {}));
export var CreateAliasRequest;
(function (CreateAliasRequest) {
    CreateAliasRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateAliasRequest || (CreateAliasRequest = {}));
export var CodeSigningPolicy;
(function (CodeSigningPolicy) {
    CodeSigningPolicy["Enforce"] = "Enforce";
    CodeSigningPolicy["Warn"] = "Warn";
})(CodeSigningPolicy || (CodeSigningPolicy = {}));
export var CodeSigningPolicies;
(function (CodeSigningPolicies) {
    CodeSigningPolicies.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CodeSigningPolicies || (CodeSigningPolicies = {}));
export var CreateCodeSigningConfigRequest;
(function (CreateCodeSigningConfigRequest) {
    CreateCodeSigningConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCodeSigningConfigRequest || (CreateCodeSigningConfigRequest = {}));
export var CodeSigningConfig;
(function (CodeSigningConfig) {
    CodeSigningConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CodeSigningConfig || (CodeSigningConfig = {}));
export var CreateCodeSigningConfigResponse;
(function (CreateCodeSigningConfigResponse) {
    CreateCodeSigningConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCodeSigningConfigResponse || (CreateCodeSigningConfigResponse = {}));
export var OnFailure;
(function (OnFailure) {
    OnFailure.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OnFailure || (OnFailure = {}));
export var OnSuccess;
(function (OnSuccess) {
    OnSuccess.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OnSuccess || (OnSuccess = {}));
export var DestinationConfig;
(function (DestinationConfig) {
    DestinationConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DestinationConfig || (DestinationConfig = {}));
export var Filter;
(function (Filter) {
    Filter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Filter || (Filter = {}));
export var FilterCriteria;
(function (FilterCriteria) {
    FilterCriteria.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FilterCriteria || (FilterCriteria = {}));
export var FunctionResponseType;
(function (FunctionResponseType) {
    FunctionResponseType["ReportBatchItemFailures"] = "ReportBatchItemFailures";
})(FunctionResponseType || (FunctionResponseType = {}));
export var EndPointType;
(function (EndPointType) {
    EndPointType["KAFKA_BOOTSTRAP_SERVERS"] = "KAFKA_BOOTSTRAP_SERVERS";
})(EndPointType || (EndPointType = {}));
export var SelfManagedEventSource;
(function (SelfManagedEventSource) {
    SelfManagedEventSource.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SelfManagedEventSource || (SelfManagedEventSource = {}));
export var SourceAccessType;
(function (SourceAccessType) {
    SourceAccessType["BASIC_AUTH"] = "BASIC_AUTH";
    SourceAccessType["CLIENT_CERTIFICATE_TLS_AUTH"] = "CLIENT_CERTIFICATE_TLS_AUTH";
    SourceAccessType["SASL_SCRAM_256_AUTH"] = "SASL_SCRAM_256_AUTH";
    SourceAccessType["SASL_SCRAM_512_AUTH"] = "SASL_SCRAM_512_AUTH";
    SourceAccessType["SERVER_ROOT_CA_CERTIFICATE"] = "SERVER_ROOT_CA_CERTIFICATE";
    SourceAccessType["VIRTUAL_HOST"] = "VIRTUAL_HOST";
    SourceAccessType["VPC_SECURITY_GROUP"] = "VPC_SECURITY_GROUP";
    SourceAccessType["VPC_SUBNET"] = "VPC_SUBNET";
})(SourceAccessType || (SourceAccessType = {}));
export var SourceAccessConfiguration;
(function (SourceAccessConfiguration) {
    SourceAccessConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SourceAccessConfiguration || (SourceAccessConfiguration = {}));
export var EventSourcePosition;
(function (EventSourcePosition) {
    EventSourcePosition["AT_TIMESTAMP"] = "AT_TIMESTAMP";
    EventSourcePosition["LATEST"] = "LATEST";
    EventSourcePosition["TRIM_HORIZON"] = "TRIM_HORIZON";
})(EventSourcePosition || (EventSourcePosition = {}));
export var CreateEventSourceMappingRequest;
(function (CreateEventSourceMappingRequest) {
    CreateEventSourceMappingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateEventSourceMappingRequest || (CreateEventSourceMappingRequest = {}));
export var EventSourceMappingConfiguration;
(function (EventSourceMappingConfiguration) {
    EventSourceMappingConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EventSourceMappingConfiguration || (EventSourceMappingConfiguration = {}));
var CodeSigningConfigNotFoundException = (function (_super) {
    __extends(CodeSigningConfigNotFoundException, _super);
    function CodeSigningConfigNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "CodeSigningConfigNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "CodeSigningConfigNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CodeSigningConfigNotFoundException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return CodeSigningConfigNotFoundException;
}(__BaseException));
export { CodeSigningConfigNotFoundException };
var CodeStorageExceededException = (function (_super) {
    __extends(CodeStorageExceededException, _super);
    function CodeStorageExceededException(opts) {
        var _this = _super.call(this, __assign({ name: "CodeStorageExceededException", $fault: "client" }, opts)) || this;
        _this.name = "CodeStorageExceededException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CodeStorageExceededException.prototype);
        _this.Type = opts.Type;
        return _this;
    }
    return CodeStorageExceededException;
}(__BaseException));
export { CodeStorageExceededException };
var CodeVerificationFailedException = (function (_super) {
    __extends(CodeVerificationFailedException, _super);
    function CodeVerificationFailedException(opts) {
        var _this = _super.call(this, __assign({ name: "CodeVerificationFailedException", $fault: "client" }, opts)) || this;
        _this.name = "CodeVerificationFailedException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CodeVerificationFailedException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return CodeVerificationFailedException;
}(__BaseException));
export { CodeVerificationFailedException };
export var FunctionCode;
(function (FunctionCode) {
    FunctionCode.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.ZipFile && { ZipFile: SENSITIVE_STRING }))); };
})(FunctionCode || (FunctionCode = {}));
export var DeadLetterConfig;
(function (DeadLetterConfig) {
    DeadLetterConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeadLetterConfig || (DeadLetterConfig = {}));
export var Environment;
(function (Environment) {
    Environment.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Variables && { Variables: SENSITIVE_STRING }))); };
})(Environment || (Environment = {}));
export var EphemeralStorage;
(function (EphemeralStorage) {
    EphemeralStorage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EphemeralStorage || (EphemeralStorage = {}));
export var FileSystemConfig;
(function (FileSystemConfig) {
    FileSystemConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FileSystemConfig || (FileSystemConfig = {}));
export var ImageConfig;
(function (ImageConfig) {
    ImageConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ImageConfig || (ImageConfig = {}));
export var PackageType;
(function (PackageType) {
    PackageType["Image"] = "Image";
    PackageType["Zip"] = "Zip";
})(PackageType || (PackageType = {}));
export var Runtime;
(function (Runtime) {
    Runtime["dotnet6"] = "dotnet6";
    Runtime["dotnetcore10"] = "dotnetcore1.0";
    Runtime["dotnetcore20"] = "dotnetcore2.0";
    Runtime["dotnetcore21"] = "dotnetcore2.1";
    Runtime["dotnetcore31"] = "dotnetcore3.1";
    Runtime["go1x"] = "go1.x";
    Runtime["java11"] = "java11";
    Runtime["java8"] = "java8";
    Runtime["java8al2"] = "java8.al2";
    Runtime["nodejs"] = "nodejs";
    Runtime["nodejs10x"] = "nodejs10.x";
    Runtime["nodejs12x"] = "nodejs12.x";
    Runtime["nodejs14x"] = "nodejs14.x";
    Runtime["nodejs16x"] = "nodejs16.x";
    Runtime["nodejs43"] = "nodejs4.3";
    Runtime["nodejs43edge"] = "nodejs4.3-edge";
    Runtime["nodejs610"] = "nodejs6.10";
    Runtime["nodejs810"] = "nodejs8.10";
    Runtime["provided"] = "provided";
    Runtime["providedal2"] = "provided.al2";
    Runtime["python27"] = "python2.7";
    Runtime["python36"] = "python3.6";
    Runtime["python37"] = "python3.7";
    Runtime["python38"] = "python3.8";
    Runtime["python39"] = "python3.9";
    Runtime["ruby25"] = "ruby2.5";
    Runtime["ruby27"] = "ruby2.7";
})(Runtime || (Runtime = {}));
export var TracingMode;
(function (TracingMode) {
    TracingMode["Active"] = "Active";
    TracingMode["PassThrough"] = "PassThrough";
})(TracingMode || (TracingMode = {}));
export var TracingConfig;
(function (TracingConfig) {
    TracingConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TracingConfig || (TracingConfig = {}));
export var VpcConfig;
(function (VpcConfig) {
    VpcConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VpcConfig || (VpcConfig = {}));
export var CreateFunctionRequest;
(function (CreateFunctionRequest) {
    CreateFunctionRequest.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Code && { Code: FunctionCode.filterSensitiveLog(obj.Code) })), (obj.Environment && { Environment: Environment.filterSensitiveLog(obj.Environment) }))); };
})(CreateFunctionRequest || (CreateFunctionRequest = {}));
export var EnvironmentError;
(function (EnvironmentError) {
    EnvironmentError.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Message && { Message: SENSITIVE_STRING }))); };
})(EnvironmentError || (EnvironmentError = {}));
export var EnvironmentResponse;
(function (EnvironmentResponse) {
    EnvironmentResponse.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Variables && { Variables: SENSITIVE_STRING })), (obj.Error && { Error: EnvironmentError.filterSensitiveLog(obj.Error) }))); };
})(EnvironmentResponse || (EnvironmentResponse = {}));
export var ImageConfigError;
(function (ImageConfigError) {
    ImageConfigError.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Message && { Message: SENSITIVE_STRING }))); };
})(ImageConfigError || (ImageConfigError = {}));
export var ImageConfigResponse;
(function (ImageConfigResponse) {
    ImageConfigResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Error && { Error: ImageConfigError.filterSensitiveLog(obj.Error) }))); };
})(ImageConfigResponse || (ImageConfigResponse = {}));
export var LastUpdateStatus;
(function (LastUpdateStatus) {
    LastUpdateStatus["Failed"] = "Failed";
    LastUpdateStatus["InProgress"] = "InProgress";
    LastUpdateStatus["Successful"] = "Successful";
})(LastUpdateStatus || (LastUpdateStatus = {}));
export var LastUpdateStatusReasonCode;
(function (LastUpdateStatusReasonCode) {
    LastUpdateStatusReasonCode["EniLimitExceeded"] = "EniLimitExceeded";
    LastUpdateStatusReasonCode["ImageAccessDenied"] = "ImageAccessDenied";
    LastUpdateStatusReasonCode["ImageDeleted"] = "ImageDeleted";
    LastUpdateStatusReasonCode["InsufficientRolePermissions"] = "InsufficientRolePermissions";
    LastUpdateStatusReasonCode["InternalError"] = "InternalError";
    LastUpdateStatusReasonCode["InvalidConfiguration"] = "InvalidConfiguration";
    LastUpdateStatusReasonCode["InvalidImage"] = "InvalidImage";
    LastUpdateStatusReasonCode["InvalidSecurityGroup"] = "InvalidSecurityGroup";
    LastUpdateStatusReasonCode["InvalidSubnet"] = "InvalidSubnet";
    LastUpdateStatusReasonCode["SubnetOutOfIPAddresses"] = "SubnetOutOfIPAddresses";
})(LastUpdateStatusReasonCode || (LastUpdateStatusReasonCode = {}));
export var Layer;
(function (Layer) {
    Layer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Layer || (Layer = {}));
export var State;
(function (State) {
    State["Active"] = "Active";
    State["Failed"] = "Failed";
    State["Inactive"] = "Inactive";
    State["Pending"] = "Pending";
})(State || (State = {}));
export var StateReasonCode;
(function (StateReasonCode) {
    StateReasonCode["Creating"] = "Creating";
    StateReasonCode["EniLimitExceeded"] = "EniLimitExceeded";
    StateReasonCode["Idle"] = "Idle";
    StateReasonCode["ImageAccessDenied"] = "ImageAccessDenied";
    StateReasonCode["ImageDeleted"] = "ImageDeleted";
    StateReasonCode["InsufficientRolePermissions"] = "InsufficientRolePermissions";
    StateReasonCode["InternalError"] = "InternalError";
    StateReasonCode["InvalidConfiguration"] = "InvalidConfiguration";
    StateReasonCode["InvalidImage"] = "InvalidImage";
    StateReasonCode["InvalidSecurityGroup"] = "InvalidSecurityGroup";
    StateReasonCode["InvalidSubnet"] = "InvalidSubnet";
    StateReasonCode["Restoring"] = "Restoring";
    StateReasonCode["SubnetOutOfIPAddresses"] = "SubnetOutOfIPAddresses";
})(StateReasonCode || (StateReasonCode = {}));
export var TracingConfigResponse;
(function (TracingConfigResponse) {
    TracingConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TracingConfigResponse || (TracingConfigResponse = {}));
export var VpcConfigResponse;
(function (VpcConfigResponse) {
    VpcConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VpcConfigResponse || (VpcConfigResponse = {}));
export var FunctionConfiguration;
(function (FunctionConfiguration) {
    FunctionConfiguration.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Environment && { Environment: EnvironmentResponse.filterSensitiveLog(obj.Environment) })), (obj.ImageConfigResponse && {
        ImageConfigResponse: ImageConfigResponse.filterSensitiveLog(obj.ImageConfigResponse),
    }))); };
})(FunctionConfiguration || (FunctionConfiguration = {}));
var InvalidCodeSignatureException = (function (_super) {
    __extends(InvalidCodeSignatureException, _super);
    function InvalidCodeSignatureException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidCodeSignatureException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidCodeSignatureException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidCodeSignatureException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return InvalidCodeSignatureException;
}(__BaseException));
export { InvalidCodeSignatureException };
export var Cors;
(function (Cors) {
    Cors.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Cors || (Cors = {}));
export var CreateFunctionUrlConfigRequest;
(function (CreateFunctionUrlConfigRequest) {
    CreateFunctionUrlConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateFunctionUrlConfigRequest || (CreateFunctionUrlConfigRequest = {}));
export var CreateFunctionUrlConfigResponse;
(function (CreateFunctionUrlConfigResponse) {
    CreateFunctionUrlConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateFunctionUrlConfigResponse || (CreateFunctionUrlConfigResponse = {}));
export var DeleteAliasRequest;
(function (DeleteAliasRequest) {
    DeleteAliasRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteAliasRequest || (DeleteAliasRequest = {}));
export var DeleteCodeSigningConfigRequest;
(function (DeleteCodeSigningConfigRequest) {
    DeleteCodeSigningConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCodeSigningConfigRequest || (DeleteCodeSigningConfigRequest = {}));
export var DeleteCodeSigningConfigResponse;
(function (DeleteCodeSigningConfigResponse) {
    DeleteCodeSigningConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCodeSigningConfigResponse || (DeleteCodeSigningConfigResponse = {}));
export var DeleteEventSourceMappingRequest;
(function (DeleteEventSourceMappingRequest) {
    DeleteEventSourceMappingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteEventSourceMappingRequest || (DeleteEventSourceMappingRequest = {}));
var ResourceInUseException = (function (_super) {
    __extends(ResourceInUseException, _super);
    function ResourceInUseException(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceInUseException", $fault: "client" }, opts)) || this;
        _this.name = "ResourceInUseException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceInUseException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return ResourceInUseException;
}(__BaseException));
export { ResourceInUseException };
export var DeleteFunctionRequest;
(function (DeleteFunctionRequest) {
    DeleteFunctionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteFunctionRequest || (DeleteFunctionRequest = {}));
export var DeleteFunctionCodeSigningConfigRequest;
(function (DeleteFunctionCodeSigningConfigRequest) {
    DeleteFunctionCodeSigningConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteFunctionCodeSigningConfigRequest || (DeleteFunctionCodeSigningConfigRequest = {}));
export var DeleteFunctionConcurrencyRequest;
(function (DeleteFunctionConcurrencyRequest) {
    DeleteFunctionConcurrencyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteFunctionConcurrencyRequest || (DeleteFunctionConcurrencyRequest = {}));
export var DeleteFunctionEventInvokeConfigRequest;
(function (DeleteFunctionEventInvokeConfigRequest) {
    DeleteFunctionEventInvokeConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteFunctionEventInvokeConfigRequest || (DeleteFunctionEventInvokeConfigRequest = {}));
export var DeleteFunctionUrlConfigRequest;
(function (DeleteFunctionUrlConfigRequest) {
    DeleteFunctionUrlConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteFunctionUrlConfigRequest || (DeleteFunctionUrlConfigRequest = {}));
export var DeleteLayerVersionRequest;
(function (DeleteLayerVersionRequest) {
    DeleteLayerVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteLayerVersionRequest || (DeleteLayerVersionRequest = {}));
export var DeleteProvisionedConcurrencyConfigRequest;
(function (DeleteProvisionedConcurrencyConfigRequest) {
    DeleteProvisionedConcurrencyConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteProvisionedConcurrencyConfigRequest || (DeleteProvisionedConcurrencyConfigRequest = {}));
export var GetAccountSettingsRequest;
(function (GetAccountSettingsRequest) {
    GetAccountSettingsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetAccountSettingsRequest || (GetAccountSettingsRequest = {}));
export var GetAccountSettingsResponse;
(function (GetAccountSettingsResponse) {
    GetAccountSettingsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetAccountSettingsResponse || (GetAccountSettingsResponse = {}));
export var GetAliasRequest;
(function (GetAliasRequest) {
    GetAliasRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetAliasRequest || (GetAliasRequest = {}));
export var GetCodeSigningConfigRequest;
(function (GetCodeSigningConfigRequest) {
    GetCodeSigningConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCodeSigningConfigRequest || (GetCodeSigningConfigRequest = {}));
export var GetCodeSigningConfigResponse;
(function (GetCodeSigningConfigResponse) {
    GetCodeSigningConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCodeSigningConfigResponse || (GetCodeSigningConfigResponse = {}));
export var GetEventSourceMappingRequest;
(function (GetEventSourceMappingRequest) {
    GetEventSourceMappingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetEventSourceMappingRequest || (GetEventSourceMappingRequest = {}));
export var GetFunctionRequest;
(function (GetFunctionRequest) {
    GetFunctionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionRequest || (GetFunctionRequest = {}));
export var FunctionCodeLocation;
(function (FunctionCodeLocation) {
    FunctionCodeLocation.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FunctionCodeLocation || (FunctionCodeLocation = {}));
export var Concurrency;
(function (Concurrency) {
    Concurrency.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Concurrency || (Concurrency = {}));
export var GetFunctionResponse;
(function (GetFunctionResponse) {
    GetFunctionResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Configuration && { Configuration: FunctionConfiguration.filterSensitiveLog(obj.Configuration) }))); };
})(GetFunctionResponse || (GetFunctionResponse = {}));
export var GetFunctionCodeSigningConfigRequest;
(function (GetFunctionCodeSigningConfigRequest) {
    GetFunctionCodeSigningConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionCodeSigningConfigRequest || (GetFunctionCodeSigningConfigRequest = {}));
export var GetFunctionCodeSigningConfigResponse;
(function (GetFunctionCodeSigningConfigResponse) {
    GetFunctionCodeSigningConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionCodeSigningConfigResponse || (GetFunctionCodeSigningConfigResponse = {}));
export var GetFunctionConcurrencyRequest;
(function (GetFunctionConcurrencyRequest) {
    GetFunctionConcurrencyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionConcurrencyRequest || (GetFunctionConcurrencyRequest = {}));
export var GetFunctionConcurrencyResponse;
(function (GetFunctionConcurrencyResponse) {
    GetFunctionConcurrencyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionConcurrencyResponse || (GetFunctionConcurrencyResponse = {}));
export var GetFunctionConfigurationRequest;
(function (GetFunctionConfigurationRequest) {
    GetFunctionConfigurationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionConfigurationRequest || (GetFunctionConfigurationRequest = {}));
export var FunctionEventInvokeConfig;
(function (FunctionEventInvokeConfig) {
    FunctionEventInvokeConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FunctionEventInvokeConfig || (FunctionEventInvokeConfig = {}));
export var GetFunctionEventInvokeConfigRequest;
(function (GetFunctionEventInvokeConfigRequest) {
    GetFunctionEventInvokeConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionEventInvokeConfigRequest || (GetFunctionEventInvokeConfigRequest = {}));
export var GetFunctionUrlConfigRequest;
(function (GetFunctionUrlConfigRequest) {
    GetFunctionUrlConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionUrlConfigRequest || (GetFunctionUrlConfigRequest = {}));
export var GetFunctionUrlConfigResponse;
(function (GetFunctionUrlConfigResponse) {
    GetFunctionUrlConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFunctionUrlConfigResponse || (GetFunctionUrlConfigResponse = {}));
export var GetLayerVersionRequest;
(function (GetLayerVersionRequest) {
    GetLayerVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLayerVersionRequest || (GetLayerVersionRequest = {}));
export var LayerVersionContentOutput;
(function (LayerVersionContentOutput) {
    LayerVersionContentOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LayerVersionContentOutput || (LayerVersionContentOutput = {}));
export var GetLayerVersionResponse;
(function (GetLayerVersionResponse) {
    GetLayerVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLayerVersionResponse || (GetLayerVersionResponse = {}));
export var GetLayerVersionByArnRequest;
(function (GetLayerVersionByArnRequest) {
    GetLayerVersionByArnRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLayerVersionByArnRequest || (GetLayerVersionByArnRequest = {}));
export var GetLayerVersionPolicyRequest;
(function (GetLayerVersionPolicyRequest) {
    GetLayerVersionPolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLayerVersionPolicyRequest || (GetLayerVersionPolicyRequest = {}));
export var GetLayerVersionPolicyResponse;
(function (GetLayerVersionPolicyResponse) {
    GetLayerVersionPolicyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLayerVersionPolicyResponse || (GetLayerVersionPolicyResponse = {}));
export var GetPolicyRequest;
(function (GetPolicyRequest) {
    GetPolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPolicyRequest || (GetPolicyRequest = {}));
export var GetPolicyResponse;
(function (GetPolicyResponse) {
    GetPolicyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPolicyResponse || (GetPolicyResponse = {}));
export var GetProvisionedConcurrencyConfigRequest;
(function (GetProvisionedConcurrencyConfigRequest) {
    GetProvisionedConcurrencyConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetProvisionedConcurrencyConfigRequest || (GetProvisionedConcurrencyConfigRequest = {}));
export var ProvisionedConcurrencyStatusEnum;
(function (ProvisionedConcurrencyStatusEnum) {
    ProvisionedConcurrencyStatusEnum["FAILED"] = "FAILED";
    ProvisionedConcurrencyStatusEnum["IN_PROGRESS"] = "IN_PROGRESS";
    ProvisionedConcurrencyStatusEnum["READY"] = "READY";
})(ProvisionedConcurrencyStatusEnum || (ProvisionedConcurrencyStatusEnum = {}));
export var GetProvisionedConcurrencyConfigResponse;
(function (GetProvisionedConcurrencyConfigResponse) {
    GetProvisionedConcurrencyConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetProvisionedConcurrencyConfigResponse || (GetProvisionedConcurrencyConfigResponse = {}));
var ProvisionedConcurrencyConfigNotFoundException = (function (_super) {
    __extends(ProvisionedConcurrencyConfigNotFoundException, _super);
    function ProvisionedConcurrencyConfigNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "ProvisionedConcurrencyConfigNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "ProvisionedConcurrencyConfigNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ProvisionedConcurrencyConfigNotFoundException.prototype);
        _this.Type = opts.Type;
        return _this;
    }
    return ProvisionedConcurrencyConfigNotFoundException;
}(__BaseException));
export { ProvisionedConcurrencyConfigNotFoundException };
var EC2AccessDeniedException = (function (_super) {
    __extends(EC2AccessDeniedException, _super);
    function EC2AccessDeniedException(opts) {
        var _this = _super.call(this, __assign({ name: "EC2AccessDeniedException", $fault: "server" }, opts)) || this;
        _this.name = "EC2AccessDeniedException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, EC2AccessDeniedException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return EC2AccessDeniedException;
}(__BaseException));
export { EC2AccessDeniedException };
var EC2ThrottledException = (function (_super) {
    __extends(EC2ThrottledException, _super);
    function EC2ThrottledException(opts) {
        var _this = _super.call(this, __assign({ name: "EC2ThrottledException", $fault: "server" }, opts)) || this;
        _this.name = "EC2ThrottledException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, EC2ThrottledException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return EC2ThrottledException;
}(__BaseException));
export { EC2ThrottledException };
var EC2UnexpectedException = (function (_super) {
    __extends(EC2UnexpectedException, _super);
    function EC2UnexpectedException(opts) {
        var _this = _super.call(this, __assign({ name: "EC2UnexpectedException", $fault: "server" }, opts)) || this;
        _this.name = "EC2UnexpectedException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, EC2UnexpectedException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        _this.EC2ErrorCode = opts.EC2ErrorCode;
        return _this;
    }
    return EC2UnexpectedException;
}(__BaseException));
export { EC2UnexpectedException };
var EFSIOException = (function (_super) {
    __extends(EFSIOException, _super);
    function EFSIOException(opts) {
        var _this = _super.call(this, __assign({ name: "EFSIOException", $fault: "client" }, opts)) || this;
        _this.name = "EFSIOException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, EFSIOException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return EFSIOException;
}(__BaseException));
export { EFSIOException };
var EFSMountConnectivityException = (function (_super) {
    __extends(EFSMountConnectivityException, _super);
    function EFSMountConnectivityException(opts) {
        var _this = _super.call(this, __assign({ name: "EFSMountConnectivityException", $fault: "client" }, opts)) || this;
        _this.name = "EFSMountConnectivityException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, EFSMountConnectivityException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return EFSMountConnectivityException;
}(__BaseException));
export { EFSMountConnectivityException };
var EFSMountFailureException = (function (_super) {
    __extends(EFSMountFailureException, _super);
    function EFSMountFailureException(opts) {
        var _this = _super.call(this, __assign({ name: "EFSMountFailureException", $fault: "client" }, opts)) || this;
        _this.name = "EFSMountFailureException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, EFSMountFailureException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return EFSMountFailureException;
}(__BaseException));
export { EFSMountFailureException };
var EFSMountTimeoutException = (function (_super) {
    __extends(EFSMountTimeoutException, _super);
    function EFSMountTimeoutException(opts) {
        var _this = _super.call(this, __assign({ name: "EFSMountTimeoutException", $fault: "client" }, opts)) || this;
        _this.name = "EFSMountTimeoutException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, EFSMountTimeoutException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return EFSMountTimeoutException;
}(__BaseException));
export { EFSMountTimeoutException };
var ENILimitReachedException = (function (_super) {
    __extends(ENILimitReachedException, _super);
    function ENILimitReachedException(opts) {
        var _this = _super.call(this, __assign({ name: "ENILimitReachedException", $fault: "server" }, opts)) || this;
        _this.name = "ENILimitReachedException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, ENILimitReachedException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return ENILimitReachedException;
}(__BaseException));
export { ENILimitReachedException };
var InvalidRequestContentException = (function (_super) {
    __extends(InvalidRequestContentException, _super);
    function InvalidRequestContentException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidRequestContentException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidRequestContentException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidRequestContentException.prototype);
        _this.Type = opts.Type;
        return _this;
    }
    return InvalidRequestContentException;
}(__BaseException));
export { InvalidRequestContentException };
var InvalidRuntimeException = (function (_super) {
    __extends(InvalidRuntimeException, _super);
    function InvalidRuntimeException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidRuntimeException", $fault: "server" }, opts)) || this;
        _this.name = "InvalidRuntimeException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, InvalidRuntimeException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return InvalidRuntimeException;
}(__BaseException));
export { InvalidRuntimeException };
var InvalidSecurityGroupIDException = (function (_super) {
    __extends(InvalidSecurityGroupIDException, _super);
    function InvalidSecurityGroupIDException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidSecurityGroupIDException", $fault: "server" }, opts)) || this;
        _this.name = "InvalidSecurityGroupIDException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, InvalidSecurityGroupIDException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return InvalidSecurityGroupIDException;
}(__BaseException));
export { InvalidSecurityGroupIDException };
var InvalidSubnetIDException = (function (_super) {
    __extends(InvalidSubnetIDException, _super);
    function InvalidSubnetIDException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidSubnetIDException", $fault: "server" }, opts)) || this;
        _this.name = "InvalidSubnetIDException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, InvalidSubnetIDException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return InvalidSubnetIDException;
}(__BaseException));
export { InvalidSubnetIDException };
var InvalidZipFileException = (function (_super) {
    __extends(InvalidZipFileException, _super);
    function InvalidZipFileException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidZipFileException", $fault: "server" }, opts)) || this;
        _this.name = "InvalidZipFileException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, InvalidZipFileException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return InvalidZipFileException;
}(__BaseException));
export { InvalidZipFileException };
export var InvocationType;
(function (InvocationType) {
    InvocationType["DryRun"] = "DryRun";
    InvocationType["Event"] = "Event";
    InvocationType["RequestResponse"] = "RequestResponse";
})(InvocationType || (InvocationType = {}));
export var LogType;
(function (LogType) {
    LogType["None"] = "None";
    LogType["Tail"] = "Tail";
})(LogType || (LogType = {}));
export var InvocationRequest;
(function (InvocationRequest) {
    InvocationRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Payload && { Payload: SENSITIVE_STRING }))); };
})(InvocationRequest || (InvocationRequest = {}));
export var InvocationResponse;
(function (InvocationResponse) {
    InvocationResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Payload && { Payload: SENSITIVE_STRING }))); };
})(InvocationResponse || (InvocationResponse = {}));
var KMSAccessDeniedException = (function (_super) {
    __extends(KMSAccessDeniedException, _super);
    function KMSAccessDeniedException(opts) {
        var _this = _super.call(this, __assign({ name: "KMSAccessDeniedException", $fault: "server" }, opts)) || this;
        _this.name = "KMSAccessDeniedException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, KMSAccessDeniedException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return KMSAccessDeniedException;
}(__BaseException));
export { KMSAccessDeniedException };
var KMSDisabledException = (function (_super) {
    __extends(KMSDisabledException, _super);
    function KMSDisabledException(opts) {
        var _this = _super.call(this, __assign({ name: "KMSDisabledException", $fault: "server" }, opts)) || this;
        _this.name = "KMSDisabledException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, KMSDisabledException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return KMSDisabledException;
}(__BaseException));
export { KMSDisabledException };
var KMSInvalidStateException = (function (_super) {
    __extends(KMSInvalidStateException, _super);
    function KMSInvalidStateException(opts) {
        var _this = _super.call(this, __assign({ name: "KMSInvalidStateException", $fault: "server" }, opts)) || this;
        _this.name = "KMSInvalidStateException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, KMSInvalidStateException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return KMSInvalidStateException;
}(__BaseException));
export { KMSInvalidStateException };
var KMSNotFoundException = (function (_super) {
    __extends(KMSNotFoundException, _super);
    function KMSNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "KMSNotFoundException", $fault: "server" }, opts)) || this;
        _this.name = "KMSNotFoundException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, KMSNotFoundException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return KMSNotFoundException;
}(__BaseException));
export { KMSNotFoundException };
var RequestTooLargeException = (function (_super) {
    __extends(RequestTooLargeException, _super);
    function RequestTooLargeException(opts) {
        var _this = _super.call(this, __assign({ name: "RequestTooLargeException", $fault: "client" }, opts)) || this;
        _this.name = "RequestTooLargeException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, RequestTooLargeException.prototype);
        _this.Type = opts.Type;
        return _this;
    }
    return RequestTooLargeException;
}(__BaseException));
export { RequestTooLargeException };
var ResourceNotReadyException = (function (_super) {
    __extends(ResourceNotReadyException, _super);
    function ResourceNotReadyException(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceNotReadyException", $fault: "server" }, opts)) || this;
        _this.name = "ResourceNotReadyException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, ResourceNotReadyException.prototype);
        _this.Type = opts.Type;
        return _this;
    }
    return ResourceNotReadyException;
}(__BaseException));
export { ResourceNotReadyException };
var SubnetIPAddressLimitReachedException = (function (_super) {
    __extends(SubnetIPAddressLimitReachedException, _super);
    function SubnetIPAddressLimitReachedException(opts) {
        var _this = _super.call(this, __assign({ name: "SubnetIPAddressLimitReachedException", $fault: "server" }, opts)) || this;
        _this.name = "SubnetIPAddressLimitReachedException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, SubnetIPAddressLimitReachedException.prototype);
        _this.Type = opts.Type;
        _this.Message = opts.Message;
        return _this;
    }
    return SubnetIPAddressLimitReachedException;
}(__BaseException));
export { SubnetIPAddressLimitReachedException };
var UnsupportedMediaTypeException = (function (_super) {
    __extends(UnsupportedMediaTypeException, _super);
    function UnsupportedMediaTypeException(opts) {
        var _this = _super.call(this, __assign({ name: "UnsupportedMediaTypeException", $fault: "client" }, opts)) || this;
        _this.name = "UnsupportedMediaTypeException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, UnsupportedMediaTypeException.prototype);
        _this.Type = opts.Type;
        return _this;
    }
    return UnsupportedMediaTypeException;
}(__BaseException));
export { UnsupportedMediaTypeException };
export var InvokeAsyncRequest;
(function (InvokeAsyncRequest) {
    InvokeAsyncRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvokeAsyncRequest || (InvokeAsyncRequest = {}));
export var InvokeAsyncResponse;
(function (InvokeAsyncResponse) {
    InvokeAsyncResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvokeAsyncResponse || (InvokeAsyncResponse = {}));
export var ListAliasesRequest;
(function (ListAliasesRequest) {
    ListAliasesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAliasesRequest || (ListAliasesRequest = {}));
export var ListAliasesResponse;
(function (ListAliasesResponse) {
    ListAliasesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAliasesResponse || (ListAliasesResponse = {}));
export var ListCodeSigningConfigsRequest;
(function (ListCodeSigningConfigsRequest) {
    ListCodeSigningConfigsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCodeSigningConfigsRequest || (ListCodeSigningConfigsRequest = {}));
export var ListCodeSigningConfigsResponse;
(function (ListCodeSigningConfigsResponse) {
    ListCodeSigningConfigsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCodeSigningConfigsResponse || (ListCodeSigningConfigsResponse = {}));
export var ListEventSourceMappingsRequest;
(function (ListEventSourceMappingsRequest) {
    ListEventSourceMappingsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListEventSourceMappingsRequest || (ListEventSourceMappingsRequest = {}));
export var ListEventSourceMappingsResponse;
(function (ListEventSourceMappingsResponse) {
    ListEventSourceMappingsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListEventSourceMappingsResponse || (ListEventSourceMappingsResponse = {}));
export var ListFunctionEventInvokeConfigsRequest;
(function (ListFunctionEventInvokeConfigsRequest) {
    ListFunctionEventInvokeConfigsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionEventInvokeConfigsRequest || (ListFunctionEventInvokeConfigsRequest = {}));
export var ListFunctionEventInvokeConfigsResponse;
(function (ListFunctionEventInvokeConfigsResponse) {
    ListFunctionEventInvokeConfigsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionEventInvokeConfigsResponse || (ListFunctionEventInvokeConfigsResponse = {}));
export var FunctionVersion;
(function (FunctionVersion) {
    FunctionVersion["ALL"] = "ALL";
})(FunctionVersion || (FunctionVersion = {}));
export var ListFunctionsRequest;
(function (ListFunctionsRequest) {
    ListFunctionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionsRequest || (ListFunctionsRequest = {}));
export var ListFunctionsResponse;
(function (ListFunctionsResponse) {
    ListFunctionsResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Functions && { Functions: obj.Functions.map(function (item) { return FunctionConfiguration.filterSensitiveLog(item); }) }))); };
})(ListFunctionsResponse || (ListFunctionsResponse = {}));
export var ListFunctionsByCodeSigningConfigRequest;
(function (ListFunctionsByCodeSigningConfigRequest) {
    ListFunctionsByCodeSigningConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionsByCodeSigningConfigRequest || (ListFunctionsByCodeSigningConfigRequest = {}));
export var ListFunctionsByCodeSigningConfigResponse;
(function (ListFunctionsByCodeSigningConfigResponse) {
    ListFunctionsByCodeSigningConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionsByCodeSigningConfigResponse || (ListFunctionsByCodeSigningConfigResponse = {}));
export var ListFunctionUrlConfigsRequest;
(function (ListFunctionUrlConfigsRequest) {
    ListFunctionUrlConfigsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionUrlConfigsRequest || (ListFunctionUrlConfigsRequest = {}));
export var FunctionUrlConfig;
(function (FunctionUrlConfig) {
    FunctionUrlConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FunctionUrlConfig || (FunctionUrlConfig = {}));
export var ListFunctionUrlConfigsResponse;
(function (ListFunctionUrlConfigsResponse) {
    ListFunctionUrlConfigsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFunctionUrlConfigsResponse || (ListFunctionUrlConfigsResponse = {}));
export var ListLayersRequest;
(function (ListLayersRequest) {
    ListLayersRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLayersRequest || (ListLayersRequest = {}));
export var LayerVersionsListItem;
(function (LayerVersionsListItem) {
    LayerVersionsListItem.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LayerVersionsListItem || (LayerVersionsListItem = {}));
export var LayersListItem;
(function (LayersListItem) {
    LayersListItem.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LayersListItem || (LayersListItem = {}));
export var ListLayersResponse;
(function (ListLayersResponse) {
    ListLayersResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLayersResponse || (ListLayersResponse = {}));
export var ListLayerVersionsRequest;
(function (ListLayerVersionsRequest) {
    ListLayerVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLayerVersionsRequest || (ListLayerVersionsRequest = {}));
export var ListLayerVersionsResponse;
(function (ListLayerVersionsResponse) {
    ListLayerVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLayerVersionsResponse || (ListLayerVersionsResponse = {}));
export var ListProvisionedConcurrencyConfigsRequest;
(function (ListProvisionedConcurrencyConfigsRequest) {
    ListProvisionedConcurrencyConfigsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListProvisionedConcurrencyConfigsRequest || (ListProvisionedConcurrencyConfigsRequest = {}));
export var ProvisionedConcurrencyConfigListItem;
(function (ProvisionedConcurrencyConfigListItem) {
    ProvisionedConcurrencyConfigListItem.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProvisionedConcurrencyConfigListItem || (ProvisionedConcurrencyConfigListItem = {}));
export var ListProvisionedConcurrencyConfigsResponse;
(function (ListProvisionedConcurrencyConfigsResponse) {
    ListProvisionedConcurrencyConfigsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListProvisionedConcurrencyConfigsResponse || (ListProvisionedConcurrencyConfigsResponse = {}));
export var ListTagsRequest;
(function (ListTagsRequest) {
    ListTagsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsRequest || (ListTagsRequest = {}));
export var ListTagsResponse;
(function (ListTagsResponse) {
    ListTagsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsResponse || (ListTagsResponse = {}));
export var ListVersionsByFunctionRequest;
(function (ListVersionsByFunctionRequest) {
    ListVersionsByFunctionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListVersionsByFunctionRequest || (ListVersionsByFunctionRequest = {}));
export var ListVersionsByFunctionResponse;
(function (ListVersionsByFunctionResponse) {
    ListVersionsByFunctionResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Versions && { Versions: obj.Versions.map(function (item) { return FunctionConfiguration.filterSensitiveLog(item); }) }))); };
})(ListVersionsByFunctionResponse || (ListVersionsByFunctionResponse = {}));
export var LayerVersionContentInput;
(function (LayerVersionContentInput) {
    LayerVersionContentInput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.ZipFile && { ZipFile: SENSITIVE_STRING }))); };
})(LayerVersionContentInput || (LayerVersionContentInput = {}));
export var PublishLayerVersionRequest;
(function (PublishLayerVersionRequest) {
    PublishLayerVersionRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Content && { Content: LayerVersionContentInput.filterSensitiveLog(obj.Content) }))); };
})(PublishLayerVersionRequest || (PublishLayerVersionRequest = {}));
export var PublishLayerVersionResponse;
(function (PublishLayerVersionResponse) {
    PublishLayerVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PublishLayerVersionResponse || (PublishLayerVersionResponse = {}));
export var PublishVersionRequest;
(function (PublishVersionRequest) {
    PublishVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PublishVersionRequest || (PublishVersionRequest = {}));
export var PutFunctionCodeSigningConfigRequest;
(function (PutFunctionCodeSigningConfigRequest) {
    PutFunctionCodeSigningConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutFunctionCodeSigningConfigRequest || (PutFunctionCodeSigningConfigRequest = {}));
export var PutFunctionCodeSigningConfigResponse;
(function (PutFunctionCodeSigningConfigResponse) {
    PutFunctionCodeSigningConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutFunctionCodeSigningConfigResponse || (PutFunctionCodeSigningConfigResponse = {}));
export var PutFunctionConcurrencyRequest;
(function (PutFunctionConcurrencyRequest) {
    PutFunctionConcurrencyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutFunctionConcurrencyRequest || (PutFunctionConcurrencyRequest = {}));
export var PutFunctionEventInvokeConfigRequest;
(function (PutFunctionEventInvokeConfigRequest) {
    PutFunctionEventInvokeConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutFunctionEventInvokeConfigRequest || (PutFunctionEventInvokeConfigRequest = {}));
export var PutProvisionedConcurrencyConfigRequest;
(function (PutProvisionedConcurrencyConfigRequest) {
    PutProvisionedConcurrencyConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutProvisionedConcurrencyConfigRequest || (PutProvisionedConcurrencyConfigRequest = {}));
export var PutProvisionedConcurrencyConfigResponse;
(function (PutProvisionedConcurrencyConfigResponse) {
    PutProvisionedConcurrencyConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutProvisionedConcurrencyConfigResponse || (PutProvisionedConcurrencyConfigResponse = {}));
export var RemoveLayerVersionPermissionRequest;
(function (RemoveLayerVersionPermissionRequest) {
    RemoveLayerVersionPermissionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveLayerVersionPermissionRequest || (RemoveLayerVersionPermissionRequest = {}));
export var RemovePermissionRequest;
(function (RemovePermissionRequest) {
    RemovePermissionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemovePermissionRequest || (RemovePermissionRequest = {}));
export var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourceRequest || (TagResourceRequest = {}));
export var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourceRequest || (UntagResourceRequest = {}));
export var UpdateAliasRequest;
(function (UpdateAliasRequest) {
    UpdateAliasRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateAliasRequest || (UpdateAliasRequest = {}));
export var UpdateCodeSigningConfigRequest;
(function (UpdateCodeSigningConfigRequest) {
    UpdateCodeSigningConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateCodeSigningConfigRequest || (UpdateCodeSigningConfigRequest = {}));
export var UpdateCodeSigningConfigResponse;
(function (UpdateCodeSigningConfigResponse) {
    UpdateCodeSigningConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateCodeSigningConfigResponse || (UpdateCodeSigningConfigResponse = {}));
export var UpdateEventSourceMappingRequest;
(function (UpdateEventSourceMappingRequest) {
    UpdateEventSourceMappingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateEventSourceMappingRequest || (UpdateEventSourceMappingRequest = {}));
export var UpdateFunctionCodeRequest;
(function (UpdateFunctionCodeRequest) {
    UpdateFunctionCodeRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.ZipFile && { ZipFile: SENSITIVE_STRING }))); };
})(UpdateFunctionCodeRequest || (UpdateFunctionCodeRequest = {}));
export var UpdateFunctionConfigurationRequest;
(function (UpdateFunctionConfigurationRequest) {
    UpdateFunctionConfigurationRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Environment && { Environment: Environment.filterSensitiveLog(obj.Environment) }))); };
})(UpdateFunctionConfigurationRequest || (UpdateFunctionConfigurationRequest = {}));
export var UpdateFunctionEventInvokeConfigRequest;
(function (UpdateFunctionEventInvokeConfigRequest) {
    UpdateFunctionEventInvokeConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateFunctionEventInvokeConfigRequest || (UpdateFunctionEventInvokeConfigRequest = {}));
export var UpdateFunctionUrlConfigRequest;
(function (UpdateFunctionUrlConfigRequest) {
    UpdateFunctionUrlConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateFunctionUrlConfigRequest || (UpdateFunctionUrlConfigRequest = {}));
export var UpdateFunctionUrlConfigResponse;
(function (UpdateFunctionUrlConfigResponse) {
    UpdateFunctionUrlConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateFunctionUrlConfigResponse || (UpdateFunctionUrlConfigResponse = {}));
