// @aws-sdk/client-lambda/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TracingMode = exports.Runtime = exports.PackageType = exports.ImageConfig = exports.FileSystemConfig = exports.EphemeralStorage = exports.Environment = exports.DeadLetterConfig = exports.FunctionCode = exports.CodeVerificationFailedException = exports.CodeStorageExceededException = exports.CodeSigningConfigNotFoundException = exports.EventSourceMappingConfiguration = exports.CreateEventSourceMappingRequest = exports.EventSourcePosition = exports.SourceAccessConfiguration = exports.SourceAccessType = exports.SelfManagedEventSource = exports.EndPointType = exports.FunctionResponseType = exports.FilterCriteria = exports.Filter = exports.DestinationConfig = exports.OnSuccess = exports.OnFailure = exports.CreateCodeSigningConfigResponse = exports.CodeSigningConfig = exports.CreateCodeSigningConfigRequest = exports.CodeSigningPolicies = exports.CodeSigningPolicy = exports.CreateAliasRequest = exports.Architecture = exports.AllowedPublishers = exports.AliasConfiguration = exports.AliasRoutingConfiguration = exports.AddPermissionResponse = exports.AddPermissionRequest = exports.FunctionUrlAuthType = exports.TooManyRequestsException = exports.ThrottleReason = exports.ServiceException = exports.ResourceNotFoundException = exports.ResourceConflictException = exports.PreconditionFailedException = exports.PolicyLengthExceededException = exports.InvalidParameterValueException = exports.AddLayerVersionPermissionResponse = exports.AddLayerVersionPermissionRequest = exports.AccountUsage = exports.AccountLimit = void 0;
exports.GetFunctionUrlConfigResponse = exports.GetFunctionUrlConfigRequest = exports.GetFunctionEventInvokeConfigRequest = exports.FunctionEventInvokeConfig = exports.GetFunctionConfigurationRequest = exports.GetFunctionConcurrencyResponse = exports.GetFunctionConcurrencyRequest = exports.GetFunctionCodeSigningConfigResponse = exports.GetFunctionCodeSigningConfigRequest = exports.GetFunctionResponse = exports.Concurrency = exports.FunctionCodeLocation = exports.GetFunctionRequest = exports.GetEventSourceMappingRequest = exports.GetCodeSigningConfigResponse = exports.GetCodeSigningConfigRequest = exports.GetAliasRequest = exports.GetAccountSettingsResponse = exports.GetAccountSettingsRequest = exports.DeleteProvisionedConcurrencyConfigRequest = exports.DeleteLayerVersionRequest = exports.DeleteFunctionUrlConfigRequest = exports.DeleteFunctionEventInvokeConfigRequest = exports.DeleteFunctionConcurrencyRequest = exports.DeleteFunctionCodeSigningConfigRequest = exports.DeleteFunctionRequest = exports.ResourceInUseException = exports.DeleteEventSourceMappingRequest = exports.DeleteCodeSigningConfigResponse = exports.DeleteCodeSigningConfigRequest = exports.DeleteAliasRequest = exports.CreateFunctionUrlConfigResponse = exports.CreateFunctionUrlConfigRequest = exports.Cors = exports.InvalidCodeSignatureException = exports.FunctionConfiguration = exports.VpcConfigResponse = exports.TracingConfigResponse = exports.StateReasonCode = exports.State = exports.Layer = exports.LastUpdateStatusReasonCode = exports.LastUpdateStatus = exports.ImageConfigResponse = exports.ImageConfigError = exports.EnvironmentResponse = exports.EnvironmentError = exports.CreateFunctionRequest = exports.VpcConfig = exports.TracingConfig = void 0;
exports.ListFunctionsResponse = exports.ListFunctionsRequest = exports.FunctionVersion = exports.ListFunctionEventInvokeConfigsResponse = exports.ListFunctionEventInvokeConfigsRequest = exports.ListEventSourceMappingsResponse = exports.ListEventSourceMappingsRequest = exports.ListCodeSigningConfigsResponse = exports.ListCodeSigningConfigsRequest = exports.ListAliasesResponse = exports.ListAliasesRequest = exports.InvokeAsyncResponse = exports.InvokeAsyncRequest = exports.UnsupportedMediaTypeException = exports.SubnetIPAddressLimitReachedException = exports.ResourceNotReadyException = exports.RequestTooLargeException = exports.KMSNotFoundException = exports.KMSInvalidStateException = exports.KMSDisabledException = exports.KMSAccessDeniedException = exports.InvocationResponse = exports.InvocationRequest = exports.LogType = exports.InvocationType = exports.InvalidZipFileException = exports.InvalidSubnetIDException = exports.InvalidSecurityGroupIDException = exports.InvalidRuntimeException = exports.InvalidRequestContentException = exports.ENILimitReachedException = exports.EFSMountTimeoutException = exports.EFSMountFailureException = exports.EFSMountConnectivityException = exports.EFSIOException = exports.EC2UnexpectedException = exports.EC2ThrottledException = exports.EC2AccessDeniedException = exports.ProvisionedConcurrencyConfigNotFoundException = exports.GetProvisionedConcurrencyConfigResponse = exports.ProvisionedConcurrencyStatusEnum = exports.GetProvisionedConcurrencyConfigRequest = exports.GetPolicyResponse = exports.GetPolicyRequest = exports.GetLayerVersionPolicyResponse = exports.GetLayerVersionPolicyRequest = exports.GetLayerVersionByArnRequest = exports.GetLayerVersionResponse = exports.LayerVersionContentOutput = exports.GetLayerVersionRequest = void 0;
exports.UpdateFunctionUrlConfigResponse = exports.UpdateFunctionUrlConfigRequest = exports.UpdateFunctionEventInvokeConfigRequest = exports.UpdateFunctionConfigurationRequest = exports.UpdateFunctionCodeRequest = exports.UpdateEventSourceMappingRequest = exports.UpdateCodeSigningConfigResponse = exports.UpdateCodeSigningConfigRequest = exports.UpdateAliasRequest = exports.UntagResourceRequest = exports.TagResourceRequest = exports.RemovePermissionRequest = exports.RemoveLayerVersionPermissionRequest = exports.PutProvisionedConcurrencyConfigResponse = exports.PutProvisionedConcurrencyConfigRequest = exports.PutFunctionEventInvokeConfigRequest = exports.PutFunctionConcurrencyRequest = exports.PutFunctionCodeSigningConfigResponse = exports.PutFunctionCodeSigningConfigRequest = exports.PublishVersionRequest = exports.PublishLayerVersionResponse = exports.PublishLayerVersionRequest = exports.LayerVersionContentInput = exports.ListVersionsByFunctionResponse = exports.ListVersionsByFunctionRequest = exports.ListTagsResponse = exports.ListTagsRequest = exports.ListProvisionedConcurrencyConfigsResponse = exports.ProvisionedConcurrencyConfigListItem = exports.ListProvisionedConcurrencyConfigsRequest = exports.ListLayerVersionsResponse = exports.ListLayerVersionsRequest = exports.ListLayersResponse = exports.LayersListItem = exports.LayerVersionsListItem = exports.ListLayersRequest = exports.ListFunctionUrlConfigsResponse = exports.FunctionUrlConfig = exports.ListFunctionUrlConfigsRequest = exports.ListFunctionsByCodeSigningConfigResponse = exports.ListFunctionsByCodeSigningConfigRequest = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const LambdaServiceException_1 = require("./LambdaServiceException");
var AccountLimit;
(function (AccountLimit) {
    AccountLimit.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AccountLimit = exports.AccountLimit || (exports.AccountLimit = {}));
var AccountUsage;
(function (AccountUsage) {
    AccountUsage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AccountUsage = exports.AccountUsage || (exports.AccountUsage = {}));
var AddLayerVersionPermissionRequest;
(function (AddLayerVersionPermissionRequest) {
    AddLayerVersionPermissionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddLayerVersionPermissionRequest = exports.AddLayerVersionPermissionRequest || (exports.AddLayerVersionPermissionRequest = {}));
var AddLayerVersionPermissionResponse;
(function (AddLayerVersionPermissionResponse) {
    AddLayerVersionPermissionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddLayerVersionPermissionResponse = exports.AddLayerVersionPermissionResponse || (exports.AddLayerVersionPermissionResponse = {}));
class InvalidParameterValueException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "InvalidParameterValueException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidParameterValueException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidParameterValueException.prototype);
        this.Type = opts.Type;
    }
}
exports.InvalidParameterValueException = InvalidParameterValueException;
class PolicyLengthExceededException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "PolicyLengthExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "PolicyLengthExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, PolicyLengthExceededException.prototype);
        this.Type = opts.Type;
    }
}
exports.PolicyLengthExceededException = PolicyLengthExceededException;
class PreconditionFailedException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "PreconditionFailedException",
            $fault: "client",
            ...opts,
        });
        this.name = "PreconditionFailedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, PreconditionFailedException.prototype);
        this.Type = opts.Type;
    }
}
exports.PreconditionFailedException = PreconditionFailedException;
class ResourceConflictException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "ResourceConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceConflictException.prototype);
        this.Type = opts.Type;
    }
}
exports.ResourceConflictException = ResourceConflictException;
class ResourceNotFoundException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
class ServiceException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "ServiceException",
            $fault: "server",
            ...opts,
        });
        this.name = "ServiceException";
        this.$fault = "server";
        Object.setPrototypeOf(this, ServiceException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.ServiceException = ServiceException;
var ThrottleReason;
(function (ThrottleReason) {
    ThrottleReason["CallerRateLimitExceeded"] = "CallerRateLimitExceeded";
    ThrottleReason["ConcurrentInvocationLimitExceeded"] = "ConcurrentInvocationLimitExceeded";
    ThrottleReason["FunctionInvocationRateLimitExceeded"] = "FunctionInvocationRateLimitExceeded";
    ThrottleReason["ReservedFunctionConcurrentInvocationLimitExceeded"] = "ReservedFunctionConcurrentInvocationLimitExceeded";
    ThrottleReason["ReservedFunctionInvocationRateLimitExceeded"] = "ReservedFunctionInvocationRateLimitExceeded";
})(ThrottleReason = exports.ThrottleReason || (exports.ThrottleReason = {}));
class TooManyRequestsException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyRequestsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.retryAfterSeconds = opts.retryAfterSeconds;
        this.Type = opts.Type;
        this.Reason = opts.Reason;
    }
}
exports.TooManyRequestsException = TooManyRequestsException;
var FunctionUrlAuthType;
(function (FunctionUrlAuthType) {
    FunctionUrlAuthType["AWS_IAM"] = "AWS_IAM";
    FunctionUrlAuthType["NONE"] = "NONE";
})(FunctionUrlAuthType = exports.FunctionUrlAuthType || (exports.FunctionUrlAuthType = {}));
var AddPermissionRequest;
(function (AddPermissionRequest) {
    AddPermissionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddPermissionRequest = exports.AddPermissionRequest || (exports.AddPermissionRequest = {}));
var AddPermissionResponse;
(function (AddPermissionResponse) {
    AddPermissionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddPermissionResponse = exports.AddPermissionResponse || (exports.AddPermissionResponse = {}));
var AliasRoutingConfiguration;
(function (AliasRoutingConfiguration) {
    AliasRoutingConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AliasRoutingConfiguration = exports.AliasRoutingConfiguration || (exports.AliasRoutingConfiguration = {}));
var AliasConfiguration;
(function (AliasConfiguration) {
    AliasConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AliasConfiguration = exports.AliasConfiguration || (exports.AliasConfiguration = {}));
var AllowedPublishers;
(function (AllowedPublishers) {
    AllowedPublishers.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AllowedPublishers = exports.AllowedPublishers || (exports.AllowedPublishers = {}));
var Architecture;
(function (Architecture) {
    Architecture["arm64"] = "arm64";
    Architecture["x86_64"] = "x86_64";
})(Architecture = exports.Architecture || (exports.Architecture = {}));
var CreateAliasRequest;
(function (CreateAliasRequest) {
    CreateAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateAliasRequest = exports.CreateAliasRequest || (exports.CreateAliasRequest = {}));
var CodeSigningPolicy;
(function (CodeSigningPolicy) {
    CodeSigningPolicy["Enforce"] = "Enforce";
    CodeSigningPolicy["Warn"] = "Warn";
})(CodeSigningPolicy = exports.CodeSigningPolicy || (exports.CodeSigningPolicy = {}));
var CodeSigningPolicies;
(function (CodeSigningPolicies) {
    CodeSigningPolicies.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CodeSigningPolicies = exports.CodeSigningPolicies || (exports.CodeSigningPolicies = {}));
var CreateCodeSigningConfigRequest;
(function (CreateCodeSigningConfigRequest) {
    CreateCodeSigningConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCodeSigningConfigRequest = exports.CreateCodeSigningConfigRequest || (exports.CreateCodeSigningConfigRequest = {}));
var CodeSigningConfig;
(function (CodeSigningConfig) {
    CodeSigningConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CodeSigningConfig = exports.CodeSigningConfig || (exports.CodeSigningConfig = {}));
var CreateCodeSigningConfigResponse;
(function (CreateCodeSigningConfigResponse) {
    CreateCodeSigningConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCodeSigningConfigResponse = exports.CreateCodeSigningConfigResponse || (exports.CreateCodeSigningConfigResponse = {}));
var OnFailure;
(function (OnFailure) {
    OnFailure.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OnFailure = exports.OnFailure || (exports.OnFailure = {}));
var OnSuccess;
(function (OnSuccess) {
    OnSuccess.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OnSuccess = exports.OnSuccess || (exports.OnSuccess = {}));
var DestinationConfig;
(function (DestinationConfig) {
    DestinationConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DestinationConfig = exports.DestinationConfig || (exports.DestinationConfig = {}));
var Filter;
(function (Filter) {
    Filter.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Filter = exports.Filter || (exports.Filter = {}));
var FilterCriteria;
(function (FilterCriteria) {
    FilterCriteria.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FilterCriteria = exports.FilterCriteria || (exports.FilterCriteria = {}));
var FunctionResponseType;
(function (FunctionResponseType) {
    FunctionResponseType["ReportBatchItemFailures"] = "ReportBatchItemFailures";
})(FunctionResponseType = exports.FunctionResponseType || (exports.FunctionResponseType = {}));
var EndPointType;
(function (EndPointType) {
    EndPointType["KAFKA_BOOTSTRAP_SERVERS"] = "KAFKA_BOOTSTRAP_SERVERS";
})(EndPointType = exports.EndPointType || (exports.EndPointType = {}));
var SelfManagedEventSource;
(function (SelfManagedEventSource) {
    SelfManagedEventSource.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SelfManagedEventSource = exports.SelfManagedEventSource || (exports.SelfManagedEventSource = {}));
var SourceAccessType;
(function (SourceAccessType) {
    SourceAccessType["BASIC_AUTH"] = "BASIC_AUTH";
    SourceAccessType["CLIENT_CERTIFICATE_TLS_AUTH"] = "CLIENT_CERTIFICATE_TLS_AUTH";
    SourceAccessType["SASL_SCRAM_256_AUTH"] = "SASL_SCRAM_256_AUTH";
    SourceAccessType["SASL_SCRAM_512_AUTH"] = "SASL_SCRAM_512_AUTH";
    SourceAccessType["SERVER_ROOT_CA_CERTIFICATE"] = "SERVER_ROOT_CA_CERTIFICATE";
    SourceAccessType["VIRTUAL_HOST"] = "VIRTUAL_HOST";
    SourceAccessType["VPC_SECURITY_GROUP"] = "VPC_SECURITY_GROUP";
    SourceAccessType["VPC_SUBNET"] = "VPC_SUBNET";
})(SourceAccessType = exports.SourceAccessType || (exports.SourceAccessType = {}));
var SourceAccessConfiguration;
(function (SourceAccessConfiguration) {
    SourceAccessConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SourceAccessConfiguration = exports.SourceAccessConfiguration || (exports.SourceAccessConfiguration = {}));
var EventSourcePosition;
(function (EventSourcePosition) {
    EventSourcePosition["AT_TIMESTAMP"] = "AT_TIMESTAMP";
    EventSourcePosition["LATEST"] = "LATEST";
    EventSourcePosition["TRIM_HORIZON"] = "TRIM_HORIZON";
})(EventSourcePosition = exports.EventSourcePosition || (exports.EventSourcePosition = {}));
var CreateEventSourceMappingRequest;
(function (CreateEventSourceMappingRequest) {
    CreateEventSourceMappingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateEventSourceMappingRequest = exports.CreateEventSourceMappingRequest || (exports.CreateEventSourceMappingRequest = {}));
var EventSourceMappingConfiguration;
(function (EventSourceMappingConfiguration) {
    EventSourceMappingConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EventSourceMappingConfiguration = exports.EventSourceMappingConfiguration || (exports.EventSourceMappingConfiguration = {}));
class CodeSigningConfigNotFoundException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "CodeSigningConfigNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "CodeSigningConfigNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CodeSigningConfigNotFoundException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.CodeSigningConfigNotFoundException = CodeSigningConfigNotFoundException;
class CodeStorageExceededException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "CodeStorageExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "CodeStorageExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CodeStorageExceededException.prototype);
        this.Type = opts.Type;
    }
}
exports.CodeStorageExceededException = CodeStorageExceededException;
class CodeVerificationFailedException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "CodeVerificationFailedException",
            $fault: "client",
            ...opts,
        });
        this.name = "CodeVerificationFailedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CodeVerificationFailedException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.CodeVerificationFailedException = CodeVerificationFailedException;
var FunctionCode;
(function (FunctionCode) {
    FunctionCode.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.ZipFile && { ZipFile: smithy_client_1.SENSITIVE_STRING }),
    });
})(FunctionCode = exports.FunctionCode || (exports.FunctionCode = {}));
var DeadLetterConfig;
(function (DeadLetterConfig) {
    DeadLetterConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeadLetterConfig = exports.DeadLetterConfig || (exports.DeadLetterConfig = {}));
var Environment;
(function (Environment) {
    Environment.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Variables && { Variables: smithy_client_1.SENSITIVE_STRING }),
    });
})(Environment = exports.Environment || (exports.Environment = {}));
var EphemeralStorage;
(function (EphemeralStorage) {
    EphemeralStorage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EphemeralStorage = exports.EphemeralStorage || (exports.EphemeralStorage = {}));
var FileSystemConfig;
(function (FileSystemConfig) {
    FileSystemConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FileSystemConfig = exports.FileSystemConfig || (exports.FileSystemConfig = {}));
var ImageConfig;
(function (ImageConfig) {
    ImageConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ImageConfig = exports.ImageConfig || (exports.ImageConfig = {}));
var PackageType;
(function (PackageType) {
    PackageType["Image"] = "Image";
    PackageType["Zip"] = "Zip";
})(PackageType = exports.PackageType || (exports.PackageType = {}));
var Runtime;
(function (Runtime) {
    Runtime["dotnet6"] = "dotnet6";
    Runtime["dotnetcore10"] = "dotnetcore1.0";
    Runtime["dotnetcore20"] = "dotnetcore2.0";
    Runtime["dotnetcore21"] = "dotnetcore2.1";
    Runtime["dotnetcore31"] = "dotnetcore3.1";
    Runtime["go1x"] = "go1.x";
    Runtime["java11"] = "java11";
    Runtime["java8"] = "java8";
    Runtime["java8al2"] = "java8.al2";
    Runtime["nodejs"] = "nodejs";
    Runtime["nodejs10x"] = "nodejs10.x";
    Runtime["nodejs12x"] = "nodejs12.x";
    Runtime["nodejs14x"] = "nodejs14.x";
    Runtime["nodejs16x"] = "nodejs16.x";
    Runtime["nodejs43"] = "nodejs4.3";
    Runtime["nodejs43edge"] = "nodejs4.3-edge";
    Runtime["nodejs610"] = "nodejs6.10";
    Runtime["nodejs810"] = "nodejs8.10";
    Runtime["provided"] = "provided";
    Runtime["providedal2"] = "provided.al2";
    Runtime["python27"] = "python2.7";
    Runtime["python36"] = "python3.6";
    Runtime["python37"] = "python3.7";
    Runtime["python38"] = "python3.8";
    Runtime["python39"] = "python3.9";
    Runtime["ruby25"] = "ruby2.5";
    Runtime["ruby27"] = "ruby2.7";
})(Runtime = exports.Runtime || (exports.Runtime = {}));
var TracingMode;
(function (TracingMode) {
    TracingMode["Active"] = "Active";
    TracingMode["PassThrough"] = "PassThrough";
})(TracingMode = exports.TracingMode || (exports.TracingMode = {}));
var TracingConfig;
(function (TracingConfig) {
    TracingConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TracingConfig = exports.TracingConfig || (exports.TracingConfig = {}));
var VpcConfig;
(function (VpcConfig) {
    VpcConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(VpcConfig = exports.VpcConfig || (exports.VpcConfig = {}));
var CreateFunctionRequest;
(function (CreateFunctionRequest) {
    CreateFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Code && { Code: FunctionCode.filterSensitiveLog(obj.Code) }),
        ...(obj.Environment && { Environment: Environment.filterSensitiveLog(obj.Environment) }),
    });
})(CreateFunctionRequest = exports.CreateFunctionRequest || (exports.CreateFunctionRequest = {}));
var EnvironmentError;
(function (EnvironmentError) {
    EnvironmentError.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Message && { Message: smithy_client_1.SENSITIVE_STRING }),
    });
})(EnvironmentError = exports.EnvironmentError || (exports.EnvironmentError = {}));
var EnvironmentResponse;
(function (EnvironmentResponse) {
    EnvironmentResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Variables && { Variables: smithy_client_1.SENSITIVE_STRING }),
        ...(obj.Error && { Error: EnvironmentError.filterSensitiveLog(obj.Error) }),
    });
})(EnvironmentResponse = exports.EnvironmentResponse || (exports.EnvironmentResponse = {}));
var ImageConfigError;
(function (ImageConfigError) {
    ImageConfigError.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Message && { Message: smithy_client_1.SENSITIVE_STRING }),
    });
})(ImageConfigError = exports.ImageConfigError || (exports.ImageConfigError = {}));
var ImageConfigResponse;
(function (ImageConfigResponse) {
    ImageConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Error && { Error: ImageConfigError.filterSensitiveLog(obj.Error) }),
    });
})(ImageConfigResponse = exports.ImageConfigResponse || (exports.ImageConfigResponse = {}));
var LastUpdateStatus;
(function (LastUpdateStatus) {
    LastUpdateStatus["Failed"] = "Failed";
    LastUpdateStatus["InProgress"] = "InProgress";
    LastUpdateStatus["Successful"] = "Successful";
})(LastUpdateStatus = exports.LastUpdateStatus || (exports.LastUpdateStatus = {}));
var LastUpdateStatusReasonCode;
(function (LastUpdateStatusReasonCode) {
    LastUpdateStatusReasonCode["EniLimitExceeded"] = "EniLimitExceeded";
    LastUpdateStatusReasonCode["ImageAccessDenied"] = "ImageAccessDenied";
    LastUpdateStatusReasonCode["ImageDeleted"] = "ImageDeleted";
    LastUpdateStatusReasonCode["InsufficientRolePermissions"] = "InsufficientRolePermissions";
    LastUpdateStatusReasonCode["InternalError"] = "InternalError";
    LastUpdateStatusReasonCode["InvalidConfiguration"] = "InvalidConfiguration";
    LastUpdateStatusReasonCode["InvalidImage"] = "InvalidImage";
    LastUpdateStatusReasonCode["InvalidSecurityGroup"] = "InvalidSecurityGroup";
    LastUpdateStatusReasonCode["InvalidSubnet"] = "InvalidSubnet";
    LastUpdateStatusReasonCode["SubnetOutOfIPAddresses"] = "SubnetOutOfIPAddresses";
})(LastUpdateStatusReasonCode = exports.LastUpdateStatusReasonCode || (exports.LastUpdateStatusReasonCode = {}));
var Layer;
(function (Layer) {
    Layer.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Layer = exports.Layer || (exports.Layer = {}));
var State;
(function (State) {
    State["Active"] = "Active";
    State["Failed"] = "Failed";
    State["Inactive"] = "Inactive";
    State["Pending"] = "Pending";
})(State = exports.State || (exports.State = {}));
var StateReasonCode;
(function (StateReasonCode) {
    StateReasonCode["Creating"] = "Creating";
    StateReasonCode["EniLimitExceeded"] = "EniLimitExceeded";
    StateReasonCode["Idle"] = "Idle";
    StateReasonCode["ImageAccessDenied"] = "ImageAccessDenied";
    StateReasonCode["ImageDeleted"] = "ImageDeleted";
    StateReasonCode["InsufficientRolePermissions"] = "InsufficientRolePermissions";
    StateReasonCode["InternalError"] = "InternalError";
    StateReasonCode["InvalidConfiguration"] = "InvalidConfiguration";
    StateReasonCode["InvalidImage"] = "InvalidImage";
    StateReasonCode["InvalidSecurityGroup"] = "InvalidSecurityGroup";
    StateReasonCode["InvalidSubnet"] = "InvalidSubnet";
    StateReasonCode["Restoring"] = "Restoring";
    StateReasonCode["SubnetOutOfIPAddresses"] = "SubnetOutOfIPAddresses";
})(StateReasonCode = exports.StateReasonCode || (exports.StateReasonCode = {}));
var TracingConfigResponse;
(function (TracingConfigResponse) {
    TracingConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TracingConfigResponse = exports.TracingConfigResponse || (exports.TracingConfigResponse = {}));
var VpcConfigResponse;
(function (VpcConfigResponse) {
    VpcConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(VpcConfigResponse = exports.VpcConfigResponse || (exports.VpcConfigResponse = {}));
var FunctionConfiguration;
(function (FunctionConfiguration) {
    FunctionConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Environment && { Environment: EnvironmentResponse.filterSensitiveLog(obj.Environment) }),
        ...(obj.ImageConfigResponse && {
            ImageConfigResponse: ImageConfigResponse.filterSensitiveLog(obj.ImageConfigResponse),
        }),
    });
})(FunctionConfiguration = exports.FunctionConfiguration || (exports.FunctionConfiguration = {}));
class InvalidCodeSignatureException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "InvalidCodeSignatureException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidCodeSignatureException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidCodeSignatureException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.InvalidCodeSignatureException = InvalidCodeSignatureException;
var Cors;
(function (Cors) {
    Cors.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Cors = exports.Cors || (exports.Cors = {}));
var CreateFunctionUrlConfigRequest;
(function (CreateFunctionUrlConfigRequest) {
    CreateFunctionUrlConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateFunctionUrlConfigRequest = exports.CreateFunctionUrlConfigRequest || (exports.CreateFunctionUrlConfigRequest = {}));
var CreateFunctionUrlConfigResponse;
(function (CreateFunctionUrlConfigResponse) {
    CreateFunctionUrlConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateFunctionUrlConfigResponse = exports.CreateFunctionUrlConfigResponse || (exports.CreateFunctionUrlConfigResponse = {}));
var DeleteAliasRequest;
(function (DeleteAliasRequest) {
    DeleteAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteAliasRequest = exports.DeleteAliasRequest || (exports.DeleteAliasRequest = {}));
var DeleteCodeSigningConfigRequest;
(function (DeleteCodeSigningConfigRequest) {
    DeleteCodeSigningConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteCodeSigningConfigRequest = exports.DeleteCodeSigningConfigRequest || (exports.DeleteCodeSigningConfigRequest = {}));
var DeleteCodeSigningConfigResponse;
(function (DeleteCodeSigningConfigResponse) {
    DeleteCodeSigningConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteCodeSigningConfigResponse = exports.DeleteCodeSigningConfigResponse || (exports.DeleteCodeSigningConfigResponse = {}));
var DeleteEventSourceMappingRequest;
(function (DeleteEventSourceMappingRequest) {
    DeleteEventSourceMappingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteEventSourceMappingRequest = exports.DeleteEventSourceMappingRequest || (exports.DeleteEventSourceMappingRequest = {}));
class ResourceInUseException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "ResourceInUseException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceInUseException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceInUseException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.ResourceInUseException = ResourceInUseException;
var DeleteFunctionRequest;
(function (DeleteFunctionRequest) {
    DeleteFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteFunctionRequest = exports.DeleteFunctionRequest || (exports.DeleteFunctionRequest = {}));
var DeleteFunctionCodeSigningConfigRequest;
(function (DeleteFunctionCodeSigningConfigRequest) {
    DeleteFunctionCodeSigningConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteFunctionCodeSigningConfigRequest = exports.DeleteFunctionCodeSigningConfigRequest || (exports.DeleteFunctionCodeSigningConfigRequest = {}));
var DeleteFunctionConcurrencyRequest;
(function (DeleteFunctionConcurrencyRequest) {
    DeleteFunctionConcurrencyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteFunctionConcurrencyRequest = exports.DeleteFunctionConcurrencyRequest || (exports.DeleteFunctionConcurrencyRequest = {}));
var DeleteFunctionEventInvokeConfigRequest;
(function (DeleteFunctionEventInvokeConfigRequest) {
    DeleteFunctionEventInvokeConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteFunctionEventInvokeConfigRequest = exports.DeleteFunctionEventInvokeConfigRequest || (exports.DeleteFunctionEventInvokeConfigRequest = {}));
var DeleteFunctionUrlConfigRequest;
(function (DeleteFunctionUrlConfigRequest) {
    DeleteFunctionUrlConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteFunctionUrlConfigRequest = exports.DeleteFunctionUrlConfigRequest || (exports.DeleteFunctionUrlConfigRequest = {}));
var DeleteLayerVersionRequest;
(function (DeleteLayerVersionRequest) {
    DeleteLayerVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteLayerVersionRequest = exports.DeleteLayerVersionRequest || (exports.DeleteLayerVersionRequest = {}));
var DeleteProvisionedConcurrencyConfigRequest;
(function (DeleteProvisionedConcurrencyConfigRequest) {
    DeleteProvisionedConcurrencyConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteProvisionedConcurrencyConfigRequest = exports.DeleteProvisionedConcurrencyConfigRequest || (exports.DeleteProvisionedConcurrencyConfigRequest = {}));
var GetAccountSettingsRequest;
(function (GetAccountSettingsRequest) {
    GetAccountSettingsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAccountSettingsRequest = exports.GetAccountSettingsRequest || (exports.GetAccountSettingsRequest = {}));
var GetAccountSettingsResponse;
(function (GetAccountSettingsResponse) {
    GetAccountSettingsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAccountSettingsResponse = exports.GetAccountSettingsResponse || (exports.GetAccountSettingsResponse = {}));
var GetAliasRequest;
(function (GetAliasRequest) {
    GetAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAliasRequest = exports.GetAliasRequest || (exports.GetAliasRequest = {}));
var GetCodeSigningConfigRequest;
(function (GetCodeSigningConfigRequest) {
    GetCodeSigningConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCodeSigningConfigRequest = exports.GetCodeSigningConfigRequest || (exports.GetCodeSigningConfigRequest = {}));
var GetCodeSigningConfigResponse;
(function (GetCodeSigningConfigResponse) {
    GetCodeSigningConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCodeSigningConfigResponse = exports.GetCodeSigningConfigResponse || (exports.GetCodeSigningConfigResponse = {}));
var GetEventSourceMappingRequest;
(function (GetEventSourceMappingRequest) {
    GetEventSourceMappingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetEventSourceMappingRequest = exports.GetEventSourceMappingRequest || (exports.GetEventSourceMappingRequest = {}));
var GetFunctionRequest;
(function (GetFunctionRequest) {
    GetFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFunctionRequest = exports.GetFunctionRequest || (exports.GetFunctionRequest = {}));
var FunctionCodeLocation;
(function (FunctionCodeLocation) {
    FunctionCodeLocation.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FunctionCodeLocation = exports.FunctionCodeLocation || (exports.FunctionCodeLocation = {}));
var Concurrency;
(function (Concurrency) {
    Concurrency.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Concurrency = exports.Concurrency || (exports.Concurrency = {}));
var GetFunctionResponse;
(function (GetFunctionResponse) {
    GetFunctionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Configuration && { Configuration: FunctionConfiguration.filterSensitiveLog(obj.Configuration) }),
    });
})(GetFunctionResponse = exports.GetFunctionResponse || (exports.GetFunctionResponse = {}));
var GetFunctionCodeSigningConfigRequest;
(function (GetFunctionCodeSigningConfigRequest) {
    GetFunctionCodeSigningConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFunctionCodeSigningConfigRequest = exports.GetFunctionCodeSigningConfigRequest || (exports.GetFunctionCodeSigningConfigRequest = {}));
var GetFunctionCodeSigningConfigResponse;
(function (GetFunctionCodeSigningConfigResponse) {
    GetFunctionCodeSigningConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFunctionCodeSigningConfigResponse = exports.GetFunctionCodeSigningConfigResponse || (exports.GetFunctionCodeSigningConfigResponse = {}));
var GetFunctionConcurrencyRequest;
(function (GetFunctionConcurrencyRequest) {
    GetFunctionConcurrencyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFunctionConcurrencyRequest = exports.GetFunctionConcurrencyRequest || (exports.GetFunctionConcurrencyRequest = {}));
var GetFunctionConcurrencyResponse;
(function (GetFunctionConcurrencyResponse) {
    GetFunctionConcurrencyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFunctionConcurrencyResponse = exports.GetFunctionConcurrencyResponse || (exports.GetFunctionConcurrencyResponse = {}));
var GetFunctionConfigurationRequest;
(function (GetFunctionConfigurationRequest) {
    GetFunctionConfigurationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFunctionConfigurationRequest = exports.GetFunctionConfigurationRequest || (exports.GetFunctionConfigurationRequest = {}));
var FunctionEventInvokeConfig;
(function (FunctionEventInvokeConfig) {
    FunctionEventInvokeConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FunctionEventInvokeConfig = exports.FunctionEventInvokeConfig || (exports.FunctionEventInvokeConfig = {}));
var GetFunctionEventInvokeConfigRequest;
(function (GetFunctionEventInvokeConfigRequest) {
    GetFunctionEventInvokeConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFunctionEventInvokeConfigRequest = exports.GetFunctionEventInvokeConfigRequest || (exports.GetFunctionEventInvokeConfigRequest = {}));
var GetFunctionUrlConfigRequest;
(function (GetFunctionUrlConfigRequest) {
    GetFunctionUrlConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFunctionUrlConfigRequest = exports.GetFunctionUrlConfigRequest || (exports.GetFunctionUrlConfigRequest = {}));
var GetFunctionUrlConfigResponse;
(function (GetFunctionUrlConfigResponse) {
    GetFunctionUrlConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFunctionUrlConfigResponse = exports.GetFunctionUrlConfigResponse || (exports.GetFunctionUrlConfigResponse = {}));
var GetLayerVersionRequest;
(function (GetLayerVersionRequest) {
    GetLayerVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetLayerVersionRequest = exports.GetLayerVersionRequest || (exports.GetLayerVersionRequest = {}));
var LayerVersionContentOutput;
(function (LayerVersionContentOutput) {
    LayerVersionContentOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LayerVersionContentOutput = exports.LayerVersionContentOutput || (exports.LayerVersionContentOutput = {}));
var GetLayerVersionResponse;
(function (GetLayerVersionResponse) {
    GetLayerVersionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetLayerVersionResponse = exports.GetLayerVersionResponse || (exports.GetLayerVersionResponse = {}));
var GetLayerVersionByArnRequest;
(function (GetLayerVersionByArnRequest) {
    GetLayerVersionByArnRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetLayerVersionByArnRequest = exports.GetLayerVersionByArnRequest || (exports.GetLayerVersionByArnRequest = {}));
var GetLayerVersionPolicyRequest;
(function (GetLayerVersionPolicyRequest) {
    GetLayerVersionPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetLayerVersionPolicyRequest = exports.GetLayerVersionPolicyRequest || (exports.GetLayerVersionPolicyRequest = {}));
var GetLayerVersionPolicyResponse;
(function (GetLayerVersionPolicyResponse) {
    GetLayerVersionPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetLayerVersionPolicyResponse = exports.GetLayerVersionPolicyResponse || (exports.GetLayerVersionPolicyResponse = {}));
var GetPolicyRequest;
(function (GetPolicyRequest) {
    GetPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPolicyRequest = exports.GetPolicyRequest || (exports.GetPolicyRequest = {}));
var GetPolicyResponse;
(function (GetPolicyResponse) {
    GetPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPolicyResponse = exports.GetPolicyResponse || (exports.GetPolicyResponse = {}));
var GetProvisionedConcurrencyConfigRequest;
(function (GetProvisionedConcurrencyConfigRequest) {
    GetProvisionedConcurrencyConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetProvisionedConcurrencyConfigRequest = exports.GetProvisionedConcurrencyConfigRequest || (exports.GetProvisionedConcurrencyConfigRequest = {}));
var ProvisionedConcurrencyStatusEnum;
(function (ProvisionedConcurrencyStatusEnum) {
    ProvisionedConcurrencyStatusEnum["FAILED"] = "FAILED";
    ProvisionedConcurrencyStatusEnum["IN_PROGRESS"] = "IN_PROGRESS";
    ProvisionedConcurrencyStatusEnum["READY"] = "READY";
})(ProvisionedConcurrencyStatusEnum = exports.ProvisionedConcurrencyStatusEnum || (exports.ProvisionedConcurrencyStatusEnum = {}));
var GetProvisionedConcurrencyConfigResponse;
(function (GetProvisionedConcurrencyConfigResponse) {
    GetProvisionedConcurrencyConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetProvisionedConcurrencyConfigResponse = exports.GetProvisionedConcurrencyConfigResponse || (exports.GetProvisionedConcurrencyConfigResponse = {}));
class ProvisionedConcurrencyConfigNotFoundException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "ProvisionedConcurrencyConfigNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ProvisionedConcurrencyConfigNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ProvisionedConcurrencyConfigNotFoundException.prototype);
        this.Type = opts.Type;
    }
}
exports.ProvisionedConcurrencyConfigNotFoundException = ProvisionedConcurrencyConfigNotFoundException;
class EC2AccessDeniedException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "EC2AccessDeniedException",
            $fault: "server",
            ...opts,
        });
        this.name = "EC2AccessDeniedException";
        this.$fault = "server";
        Object.setPrototypeOf(this, EC2AccessDeniedException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.EC2AccessDeniedException = EC2AccessDeniedException;
class EC2ThrottledException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "EC2ThrottledException",
            $fault: "server",
            ...opts,
        });
        this.name = "EC2ThrottledException";
        this.$fault = "server";
        Object.setPrototypeOf(this, EC2ThrottledException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.EC2ThrottledException = EC2ThrottledException;
class EC2UnexpectedException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "EC2UnexpectedException",
            $fault: "server",
            ...opts,
        });
        this.name = "EC2UnexpectedException";
        this.$fault = "server";
        Object.setPrototypeOf(this, EC2UnexpectedException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
        this.EC2ErrorCode = opts.EC2ErrorCode;
    }
}
exports.EC2UnexpectedException = EC2UnexpectedException;
class EFSIOException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "EFSIOException",
            $fault: "client",
            ...opts,
        });
        this.name = "EFSIOException";
        this.$fault = "client";
        Object.setPrototypeOf(this, EFSIOException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.EFSIOException = EFSIOException;
class EFSMountConnectivityException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "EFSMountConnectivityException",
            $fault: "client",
            ...opts,
        });
        this.name = "EFSMountConnectivityException";
        this.$fault = "client";
        Object.setPrototypeOf(this, EFSMountConnectivityException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.EFSMountConnectivityException = EFSMountConnectivityException;
class EFSMountFailureException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "EFSMountFailureException",
            $fault: "client",
            ...opts,
        });
        this.name = "EFSMountFailureException";
        this.$fault = "client";
        Object.setPrototypeOf(this, EFSMountFailureException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.EFSMountFailureException = EFSMountFailureException;
class EFSMountTimeoutException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "EFSMountTimeoutException",
            $fault: "client",
            ...opts,
        });
        this.name = "EFSMountTimeoutException";
        this.$fault = "client";
        Object.setPrototypeOf(this, EFSMountTimeoutException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.EFSMountTimeoutException = EFSMountTimeoutException;
class ENILimitReachedException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "ENILimitReachedException",
            $fault: "server",
            ...opts,
        });
        this.name = "ENILimitReachedException";
        this.$fault = "server";
        Object.setPrototypeOf(this, ENILimitReachedException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.ENILimitReachedException = ENILimitReachedException;
class InvalidRequestContentException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "InvalidRequestContentException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidRequestContentException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidRequestContentException.prototype);
        this.Type = opts.Type;
    }
}
exports.InvalidRequestContentException = InvalidRequestContentException;
class InvalidRuntimeException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "InvalidRuntimeException",
            $fault: "server",
            ...opts,
        });
        this.name = "InvalidRuntimeException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InvalidRuntimeException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.InvalidRuntimeException = InvalidRuntimeException;
class InvalidSecurityGroupIDException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "InvalidSecurityGroupIDException",
            $fault: "server",
            ...opts,
        });
        this.name = "InvalidSecurityGroupIDException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InvalidSecurityGroupIDException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.InvalidSecurityGroupIDException = InvalidSecurityGroupIDException;
class InvalidSubnetIDException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "InvalidSubnetIDException",
            $fault: "server",
            ...opts,
        });
        this.name = "InvalidSubnetIDException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InvalidSubnetIDException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.InvalidSubnetIDException = InvalidSubnetIDException;
class InvalidZipFileException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "InvalidZipFileException",
            $fault: "server",
            ...opts,
        });
        this.name = "InvalidZipFileException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InvalidZipFileException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.InvalidZipFileException = InvalidZipFileException;
var InvocationType;
(function (InvocationType) {
    InvocationType["DryRun"] = "DryRun";
    InvocationType["Event"] = "Event";
    InvocationType["RequestResponse"] = "RequestResponse";
})(InvocationType = exports.InvocationType || (exports.InvocationType = {}));
var LogType;
(function (LogType) {
    LogType["None"] = "None";
    LogType["Tail"] = "Tail";
})(LogType = exports.LogType || (exports.LogType = {}));
var InvocationRequest;
(function (InvocationRequest) {
    InvocationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Payload && { Payload: smithy_client_1.SENSITIVE_STRING }),
    });
})(InvocationRequest = exports.InvocationRequest || (exports.InvocationRequest = {}));
var InvocationResponse;
(function (InvocationResponse) {
    InvocationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Payload && { Payload: smithy_client_1.SENSITIVE_STRING }),
    });
})(InvocationResponse = exports.InvocationResponse || (exports.InvocationResponse = {}));
class KMSAccessDeniedException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "KMSAccessDeniedException",
            $fault: "server",
            ...opts,
        });
        this.name = "KMSAccessDeniedException";
        this.$fault = "server";
        Object.setPrototypeOf(this, KMSAccessDeniedException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.KMSAccessDeniedException = KMSAccessDeniedException;
class KMSDisabledException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "KMSDisabledException",
            $fault: "server",
            ...opts,
        });
        this.name = "KMSDisabledException";
        this.$fault = "server";
        Object.setPrototypeOf(this, KMSDisabledException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.KMSDisabledException = KMSDisabledException;
class KMSInvalidStateException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "KMSInvalidStateException",
            $fault: "server",
            ...opts,
        });
        this.name = "KMSInvalidStateException";
        this.$fault = "server";
        Object.setPrototypeOf(this, KMSInvalidStateException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.KMSInvalidStateException = KMSInvalidStateException;
class KMSNotFoundException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "KMSNotFoundException",
            $fault: "server",
            ...opts,
        });
        this.name = "KMSNotFoundException";
        this.$fault = "server";
        Object.setPrototypeOf(this, KMSNotFoundException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.KMSNotFoundException = KMSNotFoundException;
class RequestTooLargeException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "RequestTooLargeException",
            $fault: "client",
            ...opts,
        });
        this.name = "RequestTooLargeException";
        this.$fault = "client";
        Object.setPrototypeOf(this, RequestTooLargeException.prototype);
        this.Type = opts.Type;
    }
}
exports.RequestTooLargeException = RequestTooLargeException;
class ResourceNotReadyException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "ResourceNotReadyException",
            $fault: "server",
            ...opts,
        });
        this.name = "ResourceNotReadyException";
        this.$fault = "server";
        Object.setPrototypeOf(this, ResourceNotReadyException.prototype);
        this.Type = opts.Type;
    }
}
exports.ResourceNotReadyException = ResourceNotReadyException;
class SubnetIPAddressLimitReachedException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "SubnetIPAddressLimitReachedException",
            $fault: "server",
            ...opts,
        });
        this.name = "SubnetIPAddressLimitReachedException";
        this.$fault = "server";
        Object.setPrototypeOf(this, SubnetIPAddressLimitReachedException.prototype);
        this.Type = opts.Type;
        this.Message = opts.Message;
    }
}
exports.SubnetIPAddressLimitReachedException = SubnetIPAddressLimitReachedException;
class UnsupportedMediaTypeException extends LambdaServiceException_1.LambdaServiceException {
    constructor(opts) {
        super({
            name: "UnsupportedMediaTypeException",
            $fault: "client",
            ...opts,
        });
        this.name = "UnsupportedMediaTypeException";
        this.$fault = "client";
        Object.setPrototypeOf(this, UnsupportedMediaTypeException.prototype);
        this.Type = opts.Type;
    }
}
exports.UnsupportedMediaTypeException = UnsupportedMediaTypeException;
var InvokeAsyncRequest;
(function (InvokeAsyncRequest) {
    InvokeAsyncRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvokeAsyncRequest = exports.InvokeAsyncRequest || (exports.InvokeAsyncRequest = {}));
var InvokeAsyncResponse;
(function (InvokeAsyncResponse) {
    InvokeAsyncResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvokeAsyncResponse = exports.InvokeAsyncResponse || (exports.InvokeAsyncResponse = {}));
var ListAliasesRequest;
(function (ListAliasesRequest) {
    ListAliasesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAliasesRequest = exports.ListAliasesRequest || (exports.ListAliasesRequest = {}));
var ListAliasesResponse;
(function (ListAliasesResponse) {
    ListAliasesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAliasesResponse = exports.ListAliasesResponse || (exports.ListAliasesResponse = {}));
var ListCodeSigningConfigsRequest;
(function (ListCodeSigningConfigsRequest) {
    ListCodeSigningConfigsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCodeSigningConfigsRequest = exports.ListCodeSigningConfigsRequest || (exports.ListCodeSigningConfigsRequest = {}));
var ListCodeSigningConfigsResponse;
(function (ListCodeSigningConfigsResponse) {
    ListCodeSigningConfigsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCodeSigningConfigsResponse = exports.ListCodeSigningConfigsResponse || (exports.ListCodeSigningConfigsResponse = {}));
var ListEventSourceMappingsRequest;
(function (ListEventSourceMappingsRequest) {
    ListEventSourceMappingsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListEventSourceMappingsRequest = exports.ListEventSourceMappingsRequest || (exports.ListEventSourceMappingsRequest = {}));
var ListEventSourceMappingsResponse;
(function (ListEventSourceMappingsResponse) {
    ListEventSourceMappingsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListEventSourceMappingsResponse = exports.ListEventSourceMappingsResponse || (exports.ListEventSourceMappingsResponse = {}));
var ListFunctionEventInvokeConfigsRequest;
(function (ListFunctionEventInvokeConfigsRequest) {
    ListFunctionEventInvokeConfigsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFunctionEventInvokeConfigsRequest = exports.ListFunctionEventInvokeConfigsRequest || (exports.ListFunctionEventInvokeConfigsRequest = {}));
var ListFunctionEventInvokeConfigsResponse;
(function (ListFunctionEventInvokeConfigsResponse) {
    ListFunctionEventInvokeConfigsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFunctionEventInvokeConfigsResponse = exports.ListFunctionEventInvokeConfigsResponse || (exports.ListFunctionEventInvokeConfigsResponse = {}));
var FunctionVersion;
(function (FunctionVersion) {
    FunctionVersion["ALL"] = "ALL";
})(FunctionVersion = exports.FunctionVersion || (exports.FunctionVersion = {}));
var ListFunctionsRequest;
(function (ListFunctionsRequest) {
    ListFunctionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFunctionsRequest = exports.ListFunctionsRequest || (exports.ListFunctionsRequest = {}));
var ListFunctionsResponse;
(function (ListFunctionsResponse) {
    ListFunctionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Functions && { Functions: obj.Functions.map((item) => FunctionConfiguration.filterSensitiveLog(item)) }),
    });
})(ListFunctionsResponse = exports.ListFunctionsResponse || (exports.ListFunctionsResponse = {}));
var ListFunctionsByCodeSigningConfigRequest;
(function (ListFunctionsByCodeSigningConfigRequest) {
    ListFunctionsByCodeSigningConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFunctionsByCodeSigningConfigRequest = exports.ListFunctionsByCodeSigningConfigRequest || (exports.ListFunctionsByCodeSigningConfigRequest = {}));
var ListFunctionsByCodeSigningConfigResponse;
(function (ListFunctionsByCodeSigningConfigResponse) {
    ListFunctionsByCodeSigningConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFunctionsByCodeSigningConfigResponse = exports.ListFunctionsByCodeSigningConfigResponse || (exports.ListFunctionsByCodeSigningConfigResponse = {}));
var ListFunctionUrlConfigsRequest;
(function (ListFunctionUrlConfigsRequest) {
    ListFunctionUrlConfigsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFunctionUrlConfigsRequest = exports.ListFunctionUrlConfigsRequest || (exports.ListFunctionUrlConfigsRequest = {}));
var FunctionUrlConfig;
(function (FunctionUrlConfig) {
    FunctionUrlConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FunctionUrlConfig = exports.FunctionUrlConfig || (exports.FunctionUrlConfig = {}));
var ListFunctionUrlConfigsResponse;
(function (ListFunctionUrlConfigsResponse) {
    ListFunctionUrlConfigsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFunctionUrlConfigsResponse = exports.ListFunctionUrlConfigsResponse || (exports.ListFunctionUrlConfigsResponse = {}));
var ListLayersRequest;
(function (ListLayersRequest) {
    ListLayersRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListLayersRequest = exports.ListLayersRequest || (exports.ListLayersRequest = {}));
var LayerVersionsListItem;
(function (LayerVersionsListItem) {
    LayerVersionsListItem.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LayerVersionsListItem = exports.LayerVersionsListItem || (exports.LayerVersionsListItem = {}));
var LayersListItem;
(function (LayersListItem) {
    LayersListItem.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LayersListItem = exports.LayersListItem || (exports.LayersListItem = {}));
var ListLayersResponse;
(function (ListLayersResponse) {
    ListLayersResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListLayersResponse = exports.ListLayersResponse || (exports.ListLayersResponse = {}));
var ListLayerVersionsRequest;
(function (ListLayerVersionsRequest) {
    ListLayerVersionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListLayerVersionsRequest = exports.ListLayerVersionsRequest || (exports.ListLayerVersionsRequest = {}));
var ListLayerVersionsResponse;
(function (ListLayerVersionsResponse) {
    ListLayerVersionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListLayerVersionsResponse = exports.ListLayerVersionsResponse || (exports.ListLayerVersionsResponse = {}));
var ListProvisionedConcurrencyConfigsRequest;
(function (ListProvisionedConcurrencyConfigsRequest) {
    ListProvisionedConcurrencyConfigsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListProvisionedConcurrencyConfigsRequest = exports.ListProvisionedConcurrencyConfigsRequest || (exports.ListProvisionedConcurrencyConfigsRequest = {}));
var ProvisionedConcurrencyConfigListItem;
(function (ProvisionedConcurrencyConfigListItem) {
    ProvisionedConcurrencyConfigListItem.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ProvisionedConcurrencyConfigListItem = exports.ProvisionedConcurrencyConfigListItem || (exports.ProvisionedConcurrencyConfigListItem = {}));
var ListProvisionedConcurrencyConfigsResponse;
(function (ListProvisionedConcurrencyConfigsResponse) {
    ListProvisionedConcurrencyConfigsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListProvisionedConcurrencyConfigsResponse = exports.ListProvisionedConcurrencyConfigsResponse || (exports.ListProvisionedConcurrencyConfigsResponse = {}));
var ListTagsRequest;
(function (ListTagsRequest) {
    ListTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsRequest = exports.ListTagsRequest || (exports.ListTagsRequest = {}));
var ListTagsResponse;
(function (ListTagsResponse) {
    ListTagsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsResponse = exports.ListTagsResponse || (exports.ListTagsResponse = {}));
var ListVersionsByFunctionRequest;
(function (ListVersionsByFunctionRequest) {
    ListVersionsByFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListVersionsByFunctionRequest = exports.ListVersionsByFunctionRequest || (exports.ListVersionsByFunctionRequest = {}));
var ListVersionsByFunctionResponse;
(function (ListVersionsByFunctionResponse) {
    ListVersionsByFunctionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Versions && { Versions: obj.Versions.map((item) => FunctionConfiguration.filterSensitiveLog(item)) }),
    });
})(ListVersionsByFunctionResponse = exports.ListVersionsByFunctionResponse || (exports.ListVersionsByFunctionResponse = {}));
var LayerVersionContentInput;
(function (LayerVersionContentInput) {
    LayerVersionContentInput.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.ZipFile && { ZipFile: smithy_client_1.SENSITIVE_STRING }),
    });
})(LayerVersionContentInput = exports.LayerVersionContentInput || (exports.LayerVersionContentInput = {}));
var PublishLayerVersionRequest;
(function (PublishLayerVersionRequest) {
    PublishLayerVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Content && { Content: LayerVersionContentInput.filterSensitiveLog(obj.Content) }),
    });
})(PublishLayerVersionRequest = exports.PublishLayerVersionRequest || (exports.PublishLayerVersionRequest = {}));
var PublishLayerVersionResponse;
(function (PublishLayerVersionResponse) {
    PublishLayerVersionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PublishLayerVersionResponse = exports.PublishLayerVersionResponse || (exports.PublishLayerVersionResponse = {}));
var PublishVersionRequest;
(function (PublishVersionRequest) {
    PublishVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PublishVersionRequest = exports.PublishVersionRequest || (exports.PublishVersionRequest = {}));
var PutFunctionCodeSigningConfigRequest;
(function (PutFunctionCodeSigningConfigRequest) {
    PutFunctionCodeSigningConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutFunctionCodeSigningConfigRequest = exports.PutFunctionCodeSigningConfigRequest || (exports.PutFunctionCodeSigningConfigRequest = {}));
var PutFunctionCodeSigningConfigResponse;
(function (PutFunctionCodeSigningConfigResponse) {
    PutFunctionCodeSigningConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutFunctionCodeSigningConfigResponse = exports.PutFunctionCodeSigningConfigResponse || (exports.PutFunctionCodeSigningConfigResponse = {}));
var PutFunctionConcurrencyRequest;
(function (PutFunctionConcurrencyRequest) {
    PutFunctionConcurrencyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutFunctionConcurrencyRequest = exports.PutFunctionConcurrencyRequest || (exports.PutFunctionConcurrencyRequest = {}));
var PutFunctionEventInvokeConfigRequest;
(function (PutFunctionEventInvokeConfigRequest) {
    PutFunctionEventInvokeConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutFunctionEventInvokeConfigRequest = exports.PutFunctionEventInvokeConfigRequest || (exports.PutFunctionEventInvokeConfigRequest = {}));
var PutProvisionedConcurrencyConfigRequest;
(function (PutProvisionedConcurrencyConfigRequest) {
    PutProvisionedConcurrencyConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutProvisionedConcurrencyConfigRequest = exports.PutProvisionedConcurrencyConfigRequest || (exports.PutProvisionedConcurrencyConfigRequest = {}));
var PutProvisionedConcurrencyConfigResponse;
(function (PutProvisionedConcurrencyConfigResponse) {
    PutProvisionedConcurrencyConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutProvisionedConcurrencyConfigResponse = exports.PutProvisionedConcurrencyConfigResponse || (exports.PutProvisionedConcurrencyConfigResponse = {}));
var RemoveLayerVersionPermissionRequest;
(function (RemoveLayerVersionPermissionRequest) {
    RemoveLayerVersionPermissionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RemoveLayerVersionPermissionRequest = exports.RemoveLayerVersionPermissionRequest || (exports.RemoveLayerVersionPermissionRequest = {}));
var RemovePermissionRequest;
(function (RemovePermissionRequest) {
    RemovePermissionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RemovePermissionRequest = exports.RemovePermissionRequest || (exports.RemovePermissionRequest = {}));
var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagResourceRequest = exports.TagResourceRequest || (exports.TagResourceRequest = {}));
var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagResourceRequest = exports.UntagResourceRequest || (exports.UntagResourceRequest = {}));
var UpdateAliasRequest;
(function (UpdateAliasRequest) {
    UpdateAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAliasRequest = exports.UpdateAliasRequest || (exports.UpdateAliasRequest = {}));
var UpdateCodeSigningConfigRequest;
(function (UpdateCodeSigningConfigRequest) {
    UpdateCodeSigningConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateCodeSigningConfigRequest = exports.UpdateCodeSigningConfigRequest || (exports.UpdateCodeSigningConfigRequest = {}));
var UpdateCodeSigningConfigResponse;
(function (UpdateCodeSigningConfigResponse) {
    UpdateCodeSigningConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateCodeSigningConfigResponse = exports.UpdateCodeSigningConfigResponse || (exports.UpdateCodeSigningConfigResponse = {}));
var UpdateEventSourceMappingRequest;
(function (UpdateEventSourceMappingRequest) {
    UpdateEventSourceMappingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateEventSourceMappingRequest = exports.UpdateEventSourceMappingRequest || (exports.UpdateEventSourceMappingRequest = {}));
var UpdateFunctionCodeRequest;
(function (UpdateFunctionCodeRequest) {
    UpdateFunctionCodeRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.ZipFile && { ZipFile: smithy_client_1.SENSITIVE_STRING }),
    });
})(UpdateFunctionCodeRequest = exports.UpdateFunctionCodeRequest || (exports.UpdateFunctionCodeRequest = {}));
var UpdateFunctionConfigurationRequest;
(function (UpdateFunctionConfigurationRequest) {
    UpdateFunctionConfigurationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Environment && { Environment: Environment.filterSensitiveLog(obj.Environment) }),
    });
})(UpdateFunctionConfigurationRequest = exports.UpdateFunctionConfigurationRequest || (exports.UpdateFunctionConfigurationRequest = {}));
var UpdateFunctionEventInvokeConfigRequest;
(function (UpdateFunctionEventInvokeConfigRequest) {
    UpdateFunctionEventInvokeConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateFunctionEventInvokeConfigRequest = exports.UpdateFunctionEventInvokeConfigRequest || (exports.UpdateFunctionEventInvokeConfigRequest = {}));
var UpdateFunctionUrlConfigRequest;
(function (UpdateFunctionUrlConfigRequest) {
    UpdateFunctionUrlConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateFunctionUrlConfigRequest = exports.UpdateFunctionUrlConfigRequest || (exports.UpdateFunctionUrlConfigRequest = {}));
var UpdateFunctionUrlConfigResponse;
(function (UpdateFunctionUrlConfigResponse) {
    UpdateFunctionUrlConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateFunctionUrlConfigResponse = exports.UpdateFunctionUrlConfigResponse || (exports.UpdateFunctionUrlConfigResponse = {}));
