// @aws-sdk/client-lambda/dist-es/commands/AddLayerVersionPermissionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AddLayerVersionPermissionRequest, AddLayerVersionPermissionResponse } from "../models/models_0";
import { deserializeAws_restJson1AddLayerVersionPermissionCommand, serializeAws_restJson1AddLayerVersionPermissionCommand, } from "../protocols/Aws_restJson1";
var AddLayerVersionPermissionCommand = (function (_super) {
    __extends(AddLayerVersionPermissionCommand, _super);
    function AddLayerVersionPermissionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AddLayerVersionPermissionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "AddLayerVersionPermissionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AddLayerVersionPermissionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AddLayerVersionPermissionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AddLayerVersionPermissionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1AddLayerVersionPermissionCommand(input, context);
    };
    AddLayerVersionPermissionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1AddLayerVersionPermissionCommand(output, context);
    };
    return AddLayerVersionPermissionCommand;
}($Command));
export { AddLayerVersionPermissionCommand };
