// @aws-sdk/client-lambda/dist-es/commands/DeleteFunctionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteFunctionRequest } from "../models/models_0";
import { deserializeAws_restJson1DeleteFunctionCommand, serializeAws_restJson1DeleteFunctionCommand, } from "../protocols/Aws_restJson1";
var DeleteFunctionCommand = (function (_super) {
    __extends(DeleteFunctionCommand, _super);
    function DeleteFunctionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteFunctionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "DeleteFunctionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteFunctionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteFunctionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteFunctionCommand(input, context);
    };
    DeleteFunctionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteFunctionCommand(output, context);
    };
    return DeleteFunctionCommand;
}($Command));
export { DeleteFunctionCommand };
