// @aws-sdk/client-lambda/dist-es/commands/DeleteFunctionCodeSigningConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteFunctionCodeSigningConfigRequest } from "../models/models_0";
import { deserializeAws_restJson1DeleteFunctionCodeSigningConfigCommand, serializeAws_restJson1DeleteFunctionCodeSigningConfigCommand, } from "../protocols/Aws_restJson1";
var DeleteFunctionCodeSigningConfigCommand = (function (_super) {
    __extends(DeleteFunctionCodeSigningConfigCommand, _super);
    function DeleteFunctionCodeSigningConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteFunctionCodeSigningConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "DeleteFunctionCodeSigningConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteFunctionCodeSigningConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteFunctionCodeSigningConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteFunctionCodeSigningConfigCommand(input, context);
    };
    DeleteFunctionCodeSigningConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteFunctionCodeSigningConfigCommand(output, context);
    };
    return DeleteFunctionCodeSigningConfigCommand;
}($Command));
export { DeleteFunctionCodeSigningConfigCommand };
