// @aws-sdk/client-lambda/dist-es/commands/DeleteLayerVersionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteLayerVersionRequest } from "../models/models_0";
import { deserializeAws_restJson1DeleteLayerVersionCommand, serializeAws_restJson1DeleteLayerVersionCommand, } from "../protocols/Aws_restJson1";
var DeleteLayerVersionCommand = (function (_super) {
    __extends(DeleteLayerVersionCommand, _super);
    function DeleteLayerVersionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteLayerVersionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "DeleteLayerVersionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteLayerVersionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteLayerVersionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteLayerVersionCommand(input, context);
    };
    DeleteLayerVersionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteLayerVersionCommand(output, context);
    };
    return DeleteLayerVersionCommand;
}($Command));
export { DeleteLayerVersionCommand };
