// @aws-sdk/client-lambda/dist-es/commands/DeleteProvisionedConcurrencyConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteProvisionedConcurrencyConfigRequest } from "../models/models_0";
import { deserializeAws_restJson1DeleteProvisionedConcurrencyConfigCommand, serializeAws_restJson1DeleteProvisionedConcurrencyConfigCommand, } from "../protocols/Aws_restJson1";
var DeleteProvisionedConcurrencyConfigCommand = (function (_super) {
    __extends(DeleteProvisionedConcurrencyConfigCommand, _super);
    function DeleteProvisionedConcurrencyConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteProvisionedConcurrencyConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "DeleteProvisionedConcurrencyConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteProvisionedConcurrencyConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteProvisionedConcurrencyConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteProvisionedConcurrencyConfigCommand(input, context);
    };
    DeleteProvisionedConcurrencyConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteProvisionedConcurrencyConfigCommand(output, context);
    };
    return DeleteProvisionedConcurrencyConfigCommand;
}($Command));
export { DeleteProvisionedConcurrencyConfigCommand };
