// @aws-sdk/client-lambda/dist-es/commands/DeleteFunctionConcurrencyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteFunctionConcurrencyRequest } from "../models/models_0";
import { deserializeAws_restJson1DeleteFunctionConcurrencyCommand, serializeAws_restJson1DeleteFunctionConcurrencyCommand, } from "../protocols/Aws_restJson1";
var DeleteFunctionConcurrencyCommand = (function (_super) {
    __extends(DeleteFunctionConcurrencyCommand, _super);
    function DeleteFunctionConcurrencyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteFunctionConcurrencyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "DeleteFunctionConcurrencyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteFunctionConcurrencyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteFunctionConcurrencyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteFunctionConcurrencyCommand(input, context);
    };
    DeleteFunctionConcurrencyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteFunctionConcurrencyCommand(output, context);
    };
    return DeleteFunctionConcurrencyCommand;
}($Command));
export { DeleteFunctionConcurrencyCommand };
