// @aws-sdk/client-lambda/dist-es/commands/GetCodeSigningConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetCodeSigningConfigRequest, GetCodeSigningConfigResponse } from "../models/models_0";
import { deserializeAws_restJson1GetCodeSigningConfigCommand, serializeAws_restJson1GetCodeSigningConfigCommand, } from "../protocols/Aws_restJson1";
var GetCodeSigningConfigCommand = (function (_super) {
    __extends(GetCodeSigningConfigCommand, _super);
    function GetCodeSigningConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetCodeSigningConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "GetCodeSigningConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetCodeSigningConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetCodeSigningConfigResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetCodeSigningConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetCodeSigningConfigCommand(input, context);
    };
    GetCodeSigningConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetCodeSigningConfigCommand(output, context);
    };
    return GetCodeSigningConfigCommand;
}($Command));
export { GetCodeSigningConfigCommand };
