// @aws-sdk/client-lambda/dist-es/commands/GetEventSourceMappingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { EventSourceMappingConfiguration, GetEventSourceMappingRequest } from "../models/models_0";
import { deserializeAws_restJson1GetEventSourceMappingCommand, serializeAws_restJson1GetEventSourceMappingCommand, } from "../protocols/Aws_restJson1";
var GetEventSourceMappingCommand = (function (_super) {
    __extends(GetEventSourceMappingCommand, _super);
    function GetEventSourceMappingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetEventSourceMappingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "GetEventSourceMappingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetEventSourceMappingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: EventSourceMappingConfiguration.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetEventSourceMappingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetEventSourceMappingCommand(input, context);
    };
    GetEventSourceMappingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetEventSourceMappingCommand(output, context);
    };
    return GetEventSourceMappingCommand;
}($Command));
export { GetEventSourceMappingCommand };
