// @aws-sdk/client-lambda/dist-es/commands/GetFunctionConfigurationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { FunctionConfiguration, GetFunctionConfigurationRequest } from "../models/models_0";
import { deserializeAws_restJson1GetFunctionConfigurationCommand, serializeAws_restJson1GetFunctionConfigurationCommand, } from "../protocols/Aws_restJson1";
var GetFunctionConfigurationCommand = (function (_super) {
    __extends(GetFunctionConfigurationCommand, _super);
    function GetFunctionConfigurationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetFunctionConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "GetFunctionConfigurationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetFunctionConfigurationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: FunctionConfiguration.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetFunctionConfigurationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetFunctionConfigurationCommand(input, context);
    };
    GetFunctionConfigurationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetFunctionConfigurationCommand(output, context);
    };
    return GetFunctionConfigurationCommand;
}($Command));
export { GetFunctionConfigurationCommand };
