// @aws-sdk/client-lambda/dist-es/commands/GetFunctionConcurrencyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetFunctionConcurrencyRequest, GetFunctionConcurrencyResponse } from "../models/models_0";
import { deserializeAws_restJson1GetFunctionConcurrencyCommand, serializeAws_restJson1GetFunctionConcurrencyCommand, } from "../protocols/Aws_restJson1";
var GetFunctionConcurrencyCommand = (function (_super) {
    __extends(GetFunctionConcurrencyCommand, _super);
    function GetFunctionConcurrencyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetFunctionConcurrencyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "GetFunctionConcurrencyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetFunctionConcurrencyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetFunctionConcurrencyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetFunctionConcurrencyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetFunctionConcurrencyCommand(input, context);
    };
    GetFunctionConcurrencyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetFunctionConcurrencyCommand(output, context);
    };
    return GetFunctionConcurrencyCommand;
}($Command));
export { GetFunctionConcurrencyCommand };
