// @aws-sdk/client-lambda/dist-es/commands/GetFunctionUrlConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetFunctionUrlConfigRequest, GetFunctionUrlConfigResponse } from "../models/models_0";
import { deserializeAws_restJson1GetFunctionUrlConfigCommand, serializeAws_restJson1GetFunctionUrlConfigCommand, } from "../protocols/Aws_restJson1";
var GetFunctionUrlConfigCommand = (function (_super) {
    __extends(GetFunctionUrlConfigCommand, _super);
    function GetFunctionUrlConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetFunctionUrlConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "GetFunctionUrlConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetFunctionUrlConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetFunctionUrlConfigResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetFunctionUrlConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetFunctionUrlConfigCommand(input, context);
    };
    GetFunctionUrlConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetFunctionUrlConfigCommand(output, context);
    };
    return GetFunctionUrlConfigCommand;
}($Command));
export { GetFunctionUrlConfigCommand };
