// @aws-sdk/client-lambda/dist-es/commands/ListEventSourceMappingsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListEventSourceMappingsRequest, ListEventSourceMappingsResponse } from "../models/models_0";
import { deserializeAws_restJson1ListEventSourceMappingsCommand, serializeAws_restJson1ListEventSourceMappingsCommand, } from "../protocols/Aws_restJson1";
var ListEventSourceMappingsCommand = (function (_super) {
    __extends(ListEventSourceMappingsCommand, _super);
    function ListEventSourceMappingsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListEventSourceMappingsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "ListEventSourceMappingsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListEventSourceMappingsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListEventSourceMappingsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListEventSourceMappingsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListEventSourceMappingsCommand(input, context);
    };
    ListEventSourceMappingsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListEventSourceMappingsCommand(output, context);
    };
    return ListEventSourceMappingsCommand;
}($Command));
export { ListEventSourceMappingsCommand };
