// @aws-sdk/client-lambda/dist-es/commands/ListFunctionUrlConfigsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListFunctionUrlConfigsRequest, ListFunctionUrlConfigsResponse } from "../models/models_0";
import { deserializeAws_restJson1ListFunctionUrlConfigsCommand, serializeAws_restJson1ListFunctionUrlConfigsCommand, } from "../protocols/Aws_restJson1";
var ListFunctionUrlConfigsCommand = (function (_super) {
    __extends(ListFunctionUrlConfigsCommand, _super);
    function ListFunctionUrlConfigsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListFunctionUrlConfigsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "ListFunctionUrlConfigsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListFunctionUrlConfigsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListFunctionUrlConfigsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListFunctionUrlConfigsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListFunctionUrlConfigsCommand(input, context);
    };
    ListFunctionUrlConfigsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListFunctionUrlConfigsCommand(output, context);
    };
    return ListFunctionUrlConfigsCommand;
}($Command));
export { ListFunctionUrlConfigsCommand };
