// @aws-sdk/client-lambda/dist-es/commands/ListProvisionedConcurrencyConfigsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListProvisionedConcurrencyConfigsRequest, ListProvisionedConcurrencyConfigsResponse, } from "../models/models_0";
import { deserializeAws_restJson1ListProvisionedConcurrencyConfigsCommand, serializeAws_restJson1ListProvisionedConcurrencyConfigsCommand, } from "../protocols/Aws_restJson1";
var ListProvisionedConcurrencyConfigsCommand = (function (_super) {
    __extends(ListProvisionedConcurrencyConfigsCommand, _super);
    function ListProvisionedConcurrencyConfigsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListProvisionedConcurrencyConfigsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "ListProvisionedConcurrencyConfigsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListProvisionedConcurrencyConfigsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListProvisionedConcurrencyConfigsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListProvisionedConcurrencyConfigsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListProvisionedConcurrencyConfigsCommand(input, context);
    };
    ListProvisionedConcurrencyConfigsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListProvisionedConcurrencyConfigsCommand(output, context);
    };
    return ListProvisionedConcurrencyConfigsCommand;
}($Command));
export { ListProvisionedConcurrencyConfigsCommand };
