// @aws-sdk/client-lambda/dist-es/commands/ListVersionsByFunctionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListVersionsByFunctionRequest, ListVersionsByFunctionResponse } from "../models/models_0";
import { deserializeAws_restJson1ListVersionsByFunctionCommand, serializeAws_restJson1ListVersionsByFunctionCommand, } from "../protocols/Aws_restJson1";
var ListVersionsByFunctionCommand = (function (_super) {
    __extends(ListVersionsByFunctionCommand, _super);
    function ListVersionsByFunctionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListVersionsByFunctionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "ListVersionsByFunctionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListVersionsByFunctionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListVersionsByFunctionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListVersionsByFunctionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListVersionsByFunctionCommand(input, context);
    };
    ListVersionsByFunctionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListVersionsByFunctionCommand(output, context);
    };
    return ListVersionsByFunctionCommand;
}($Command));
export { ListVersionsByFunctionCommand };
