// @aws-sdk/client-lambda/dist-es/commands/PutFunctionEventInvokeConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { FunctionEventInvokeConfig, PutFunctionEventInvokeConfigRequest } from "../models/models_0";
import { deserializeAws_restJson1PutFunctionEventInvokeConfigCommand, serializeAws_restJson1PutFunctionEventInvokeConfigCommand, } from "../protocols/Aws_restJson1";
var PutFunctionEventInvokeConfigCommand = (function (_super) {
    __extends(PutFunctionEventInvokeConfigCommand, _super);
    function PutFunctionEventInvokeConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutFunctionEventInvokeConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "PutFunctionEventInvokeConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutFunctionEventInvokeConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: FunctionEventInvokeConfig.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutFunctionEventInvokeConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1PutFunctionEventInvokeConfigCommand(input, context);
    };
    PutFunctionEventInvokeConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1PutFunctionEventInvokeConfigCommand(output, context);
    };
    return PutFunctionEventInvokeConfigCommand;
}($Command));
export { PutFunctionEventInvokeConfigCommand };
