// @aws-sdk/client-lambda/dist-es/commands/PutFunctionConcurrencyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { Concurrency, PutFunctionConcurrencyRequest } from "../models/models_0";
import { deserializeAws_restJson1PutFunctionConcurrencyCommand, serializeAws_restJson1PutFunctionConcurrencyCommand, } from "../protocols/Aws_restJson1";
var PutFunctionConcurrencyCommand = (function (_super) {
    __extends(PutFunctionConcurrencyCommand, _super);
    function PutFunctionConcurrencyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutFunctionConcurrencyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "PutFunctionConcurrencyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutFunctionConcurrencyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: Concurrency.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutFunctionConcurrencyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1PutFunctionConcurrencyCommand(input, context);
    };
    PutFunctionConcurrencyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1PutFunctionConcurrencyCommand(output, context);
    };
    return PutFunctionConcurrencyCommand;
}($Command));
export { PutFunctionConcurrencyCommand };
