// @aws-sdk/client-lambda/dist-es/commands/UpdateAliasCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AliasConfiguration, UpdateAliasRequest } from "../models/models_0";
import { deserializeAws_restJson1UpdateAliasCommand, serializeAws_restJson1UpdateAliasCommand, } from "../protocols/Aws_restJson1";
var UpdateAliasCommand = (function (_super) {
    __extends(UpdateAliasCommand, _super);
    function UpdateAliasCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateAliasCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "UpdateAliasCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateAliasRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AliasConfiguration.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateAliasCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateAliasCommand(input, context);
    };
    UpdateAliasCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateAliasCommand(output, context);
    };
    return UpdateAliasCommand;
}($Command));
export { UpdateAliasCommand };
