// @aws-sdk/client-lambda/dist-es/commands/UpdateEventSourceMappingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { EventSourceMappingConfiguration, UpdateEventSourceMappingRequest } from "../models/models_0";
import { deserializeAws_restJson1UpdateEventSourceMappingCommand, serializeAws_restJson1UpdateEventSourceMappingCommand, } from "../protocols/Aws_restJson1";
var UpdateEventSourceMappingCommand = (function (_super) {
    __extends(UpdateEventSourceMappingCommand, _super);
    function UpdateEventSourceMappingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateEventSourceMappingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "UpdateEventSourceMappingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateEventSourceMappingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: EventSourceMappingConfiguration.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateEventSourceMappingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateEventSourceMappingCommand(input, context);
    };
    UpdateEventSourceMappingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateEventSourceMappingCommand(output, context);
    };
    return UpdateEventSourceMappingCommand;
}($Command));
export { UpdateEventSourceMappingCommand };
